---
title: "SiT - Mouse Coronal Brain Section (CBS2) - 03.isoswitch"
author: '[Kevin Lebrigand](mailto:lebrigand@ipmc.cnrs.fr)§, Joseph Bergenstråhle §, Kim Thrane §, Annelie Mollbrink, Pascal Barbry, Rainer Waldmann and Joakim Lundeberg'
date: 'Compiled: `r format(Sys.Date(), "%B %d, %Y")`'
abstract: 'The spatial landscape of gene expression isoforms in tissue sections.'
output:
  pdf_document:
    toc: yes
    toc_depth: '3'
  html_notebook:
    code_folding: none
    theme: journal
    toc: yes
    toc_depth: 3
    toc_float: yes
  html_document:
    df_print: paged
    toc: yes
    toc_depth: '3'
---

<style type="text/css">

body, td {
   font-size: 15px;
}
code.r{
  font-size: 15px;
}
pre {
  font-size: 15px
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  cache = FALSE,
  cache.lazy = FALSE,
  tidy = TRUE
)

```

# Loading CBS2

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=4, fig.width=6}

setwd("/data/10x_data/10x_visium/000-github")
source("00.common.import.R")

CBS1 <- readRDS("./output/CBS1.nanopore.deconv.rds")
CBS2 <- readRDS("./output/CBS2.nanopore.deconv.rds")

clusters <- c("Midbrain","Hippocampus area","Isocortex-1","Isocortex-2","Olfactory area","Fiber tracts","Retrosplenial area","CA1/CA2","Thalamus","Hypothalamus","CA3","DG")

```

# CBS1/CBS2 isoform number distribution

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

df <- as.data.frame(CBS2@assays$ISO@counts)
df$geneId <- sapply(strsplit(rownames(df), "\\.\\."), `[`, 1)
df$transcriptId <- sapply(strsplit(rownames(df), "\\.\\."), `[`, 2)
df <- as.data.frame(table(sapply(strsplit(rownames(CBS2@assays$ISO@counts), "\\.\\."), `[`, 1)))
t <- as.data.frame(table(df$Freq))
t$rep <- "CBS2"

df2 <- as.data.frame(CBS1@assays$ISO@counts)
df2$geneId <- sapply(strsplit(rownames(df2), "\\.\\."), `[`, 1)
df2$transcriptId <- sapply(strsplit(rownames(df2), "\\.\\."), `[`, 2)
df2 <- as.data.frame(table(sapply(strsplit(rownames(CBS1@assays$ISO@counts), "\\.\\."), `[`, 1)))
t2 <- as.data.frame(table(df2$Freq))
t2$rep <- "CBS1"

t$pct <- round(100*t$Freq/sum(t$Freq), digits = 1)
t2$pct <- round(100*t2$Freq/sum(t2$Freq), digits = 1)
dat <- rbind(t,t2)

pdf("isoform.per.gene.pdf", width=8, height=6, useDingbats=FALSE)
ggplot(data=dat, aes(x=Var1, y=Freq, fill=rep)) +
  geom_bar(stat="identity", position=position_dodge())+
  geom_text(aes(label=Freq), vjust=-0.6, color="black",
            position = position_dodge(0.9), size=3.5)+
  scale_fill_brewer(palette="Paired")+
  theme_minimal()
dev.off()

```

# CBS2 generate MULTI assay containing only isoforms from multi-isoforms genes

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

df <- as.data.frame(CBS2@assays$ISO@counts)
df$geneId <- sapply(strsplit(rownames(df), "\\.\\."), `[`, 1)
t <- as.data.frame(table(df$geneId))
multi <- t[which(t$Freq>1),]

df.multi <- df[df$geneId %in% multi$Var1,]
df.multi <- df.multi[ , -which(names(df.multi) %in% c("geneId"))]

somme <- as.data.frame(Matrix::rowSums(df.multi))
colnames(somme) <- c("somme")
somme$cat <- ">1000"
somme$cat[which(somme$somme<=1000)] <- "500-1000"
somme$cat[which(somme$somme<500)] <- "200-500"
somme$cat[which(somme$somme<200)] <- "100-200"
somme$cat[which(somme$somme<100)] <- "50-100"
somme$cat[which(somme$somme<50)] <- "25-50"
somme$cat[which(somme$somme<25)] <- "10-25"
somme$cat[which(somme$somme<10)] <- "<10"

new_order <- with(data, reorder(variety , note, median , na.rm=T))

ggplot(somme, aes(cat)) +
  geom_bar(fill = "steelblue")

```

# CBS2 work on majoritary isoform

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

df.multi.high <- df.multi
df.multi.high$geneId <- sapply(strsplit(rownames(df.multi.high), "\\.\\."), `[`, 1)
t <- as.data.frame(table(df.multi.high$geneId))
multi <- t[which(t$Freq>1),]

df.multi.high <- df.multi.high[df.multi.high$geneId %in% multi$Var1,]
df.multi.high <- df.multi.high[ , -which(names(df.multi.high) %in% c("geneId"))]

gg <- multi$Var1
majoritary <- character()
dat <- data.frame()
for(i in 1:length(gg)){
  idx <- grep(paste0("^",gg[i],"\\.\\."),rownames(df.multi.high))
  iso <- Matrix::rowSums(df.multi.high[idx,])
  x <- max(iso)/sum(iso)
  
  majoritary <- c(majoritary, names(which.max(iso)))
  
  de <- data.frame(gg[i],max(iso),sum(iso),x)
  names(de)<-c("gene","max","total","ratiomax")
  dat <- rbind(dat, de)
  
  #print(paste0(gg[i], ",", max(iso), ",", sum(iso), ",", x, ",", sep=""))
}

hs=hist(dat$ratiomax, breaks=20)

ggplot(data=dat, aes(x=total,y=ratiomax)) +  
      geom_point(shape = 21, colour = "black", fill="lightsteelblue3") +
      theme_bw() + theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank()) + 
      scale_x_continuous(trans = 'log10')
      #geom_text_repel(data=subset(dat, dat$ratiomax < 0.5 & dat$total > 1000), aes(label=gene), size=5) +
      #geom_point(data=subset(dat, dat$ratiomax < 0.5 & dat$total > 1000), col="red") 

CBS2[["MULTI"]] <- CreateAssayObject(counts = df.multi.high)
CBS2 <- NormalizeData(object = CBS2, assay = "MULTI")
CBS2 <- scale_my_data(CBS2, assay="MULTI")

```

# CBS2 Isoform switch detection

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

DefaultAssay(object = CBS2) <- "MULTI"
total <- data.frame()
totaladj <- data.frame()
for (i in (1:11)){
  k <- i+1
  for (j in (k:12)){
    if(i != j){
      print(paste(i, " ", j, " ",clusters[i], " vs ", clusters[j], sep=""))
      
      markers <- FindMarkers(object = CBS2, ident.1=clusters[i], ident.2=clusters[j])
      markers$cluster <- clusters[j]
      markers$region.1 <- clusters[i]
      markers$region.2 <- clusters[j]
      #markers$contrast <- paste(clusters[i], "vs", clusters[j], sep=" ")
      markers[which(markers$avg_log2FC>0),]$cluster <- clusters[i]
      markers$geneId <- sapply(strsplit(rownames(markers), "\\.\\."), `[`, 1)
      markers$transcriptId <- sapply(strsplit(rownames(markers), "\\.\\."), `[`, 2)
      markers$majo <- 0
      markers$majo[which(rownames(markers) %in% majoritary)] <- 1
      markers <- subset(markers, markers$majo == 1 | (markers$p_val_adj < 0.05 & markers$majo == 0))
      all.genes <- unique(markers$geneId)
      for (k in (1:length(all.genes))){
         sub <- markers[which(markers$geneId == all.genes[k]),]
         nb.clusters <- unique(sub$cluster)
         nb.transcripts <- unique(sub$transcriptId)
        
         if(length(nb.clusters) > 1 & length(nb.transcripts) > 1){
           totaladj <- rbind(totaladj, sub)
         }
      }
      print (dim(totaladj))
      print (unique(totaladj$geneId))
    }
  }
}
write.table(totaladj, file="output/CBS2.isoswitch.new.csv", sep=",")

```

# Loading CBS1

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=4, fig.width=6}

CBS1 <- readRDS("./output/CBS1.nanopore.rds")
clusters <- c("Midbrain","Hippocampus area","Isocortex-1","Isocortex-2","Olfactory area","Fiber tracts","Retrosplenial area","CA1/CA2","Thalamus","Hypothalamus","CA3","DG")

```

# CBS1 generate MULTI assay containing only isoforms from multi-isoforms genes

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

df <- as.data.frame(CBS1@assays$ISO@counts)
df$geneId <- sapply(strsplit(rownames(df), "\\.\\."), `[`, 1)
t <- as.data.frame(table(df$geneId))
multi <- t[which(t$Freq>1),]

df.multi <- df[df$geneId %in% multi$Var1,]
df.multi <- df.multi[ , -which(names(df.multi) %in% c("geneId"))]

somme <- as.data.frame(Matrix::rowSums(df.multi))
colnames(somme) <- c("somme")
somme$cat <- ">1000"
somme$cat[which(somme$somme<=1000)] <- "500-1000"
somme$cat[which(somme$somme<500)] <- "200-500"
somme$cat[which(somme$somme<200)] <- "100-200"
somme$cat[which(somme$somme<100)] <- "50-100"
somme$cat[which(somme$somme<50)] <- "25-50"
somme$cat[which(somme$somme<25)] <- "10-25"
somme$cat[which(somme$somme<10)] <- "<10"

ggplot(somme, aes(cat)) +
  geom_bar(fill = "steelblue")

df.multi.high <- df.multi
df.multi.high$geneId <- sapply(strsplit(rownames(df.multi.high), "\\.\\."), `[`, 1)
t <- as.data.frame(table(df.multi.high$geneId))
multi <- t[which(t$Freq>1),]

df.multi.high <- df.multi.high[df.multi.high$geneId %in% multi$Var1,]
df.multi.high <- df.multi.high[ , -which(names(df.multi.high) %in% c("geneId"))]

```

# CBS1 work on majoritary isoform

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

gg <- multi$Var1
majoritary <- character()
dat <- data.frame()
for(i in 1:length(gg)){
  idx <- grep(paste0("^",gg[i],"\\.\\."),rownames(df.multi.high))
  iso <- Matrix::rowSums(df.multi.high[idx,])
  x <- max(iso)/sum(iso)
  
  majoritary <- c(majoritary, names(which.max(iso)))
  
  de <- data.frame(gg[i],max(iso),sum(iso),x)
  names(de)<-c("gene","max","total","ratiomax")
  dat <- rbind(dat, de)
  
  #print(paste0(gg[i], ",", max(iso), ",", sum(iso), ",", x, ",", sep=""))
}

hs=hist(dat$ratiomax, breaks=20)
#str(hs)
#tail(cbind(hs$mids,hs$counts),20)
#barplot(hs$counts)
#summary(hs$counts)

ggplot(data=dat, aes(x=total,y=ratiomax)) +  
      geom_point(shape = 21, colour = "black", fill="lightsteelblue3") +
      theme_bw() + theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank()) + 
      scale_x_continuous(trans = 'log10')
      #geom_text_repel(data=subset(dat, dat$ratiomax < 0.5 & dat$total > 1000), aes(label=gene), size=5) +
      #geom_point(data=subset(dat, dat$ratiomax < 0.5 & dat$total > 1000), col="red") 

CBS1[["MULTI"]] <- CreateAssayObject(counts = df.multi.high)
CBS1 <- NormalizeData(object = CBS1, assay = "MULTI")
#CBS2 <- scale_my_data(CBS2, assay="MULTI")

```

# CBS1 isoform switch detection

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

DefaultAssay(object = CBS1) <- "MULTI"
total <- data.frame()
totaladj <- data.frame()
for (i in (1:11)){
  k <- i+1
  for (j in (k:12)){
    if(i != j){
      print(paste(i, " ", j, " ",clusters[i], " vs ", clusters[j], sep=""))
      
      markers <- FindMarkers(object = CBS1, ident.1=clusters[i], ident.2=clusters[j])
      markers$cluster <- clusters[j]
      markers$region.1 <- clusters[i]
      markers$region.2 <- clusters[j]
      #markers$contrast <- paste(clusters[i], "vs", clusters[j], sep=" ")
      markers[which(markers$avg_log2FC>0),]$cluster <- clusters[i]
      markers$geneId <- sapply(strsplit(rownames(markers), "\\.\\."), `[`, 1)
      markers$transcriptId <- sapply(strsplit(rownames(markers), "\\.\\."), `[`, 2)
      markers$majo <- 0
      markers$majo[which(rownames(markers) %in% majoritary)] <- 1
      markers <- subset(markers, markers$majo == 1 | (markers$p_val_adj < 0.05 & markers$majo == 0))
      all.genes <- unique(markers$geneId)
      for (k in (1:length(all.genes))){
         sub <- markers[which(markers$geneId == all.genes[k]),]
         nb.clusters <- unique(sub$cluster)
         nb.transcripts <- unique(sub$transcriptId)
        
         if(length(nb.clusters) > 1 & length(nb.transcripts) > 1){
           totaladj <- rbind(totaladj, sub)
         }
      }
      print (dim(totaladj))
      print (unique(totaladj$geneId))
    }
  }
}

write.table(totaladj, file="output/CBS1.isoswitch.new.csv", sep=",")

```

# Loading MOB

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=4, fig.width=6}

MOB <- readRDS("./output/MOB.nanopore.rds")
clusters <- c("Granule Cell Layer (GCL+RMS)","Mitral Cell Layer (MCL)","Outer plexiform Layer (EPL)","Glomerular Layer (GL)","Olfactory Nerve Layer (ONL)")

```

# MOB generate MULTI assay containing only isoforms from multi-isoforms genes

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

df <- as.data.frame(MOB@assays$ISO@counts)
df$geneId <- sapply(strsplit(rownames(df), "\\.\\."), `[`, 1)
t <- as.data.frame(table(df$geneId))
multi <- t[which(t$Freq>1),]

df.multi <- df[df$geneId %in% multi$Var1,]
df.multi <- df.multi[ , -which(names(df.multi) %in% c("geneId"))]

somme <- as.data.frame(Matrix::rowSums(df.multi))
colnames(somme) <- c("somme")
somme$cat <- ">1000"
somme$cat[which(somme$somme<=1000)] <- "500-1000"
somme$cat[which(somme$somme<500)] <- "200-500"
somme$cat[which(somme$somme<200)] <- "100-200"
somme$cat[which(somme$somme<100)] <- "50-100"
somme$cat[which(somme$somme<50)] <- "25-50"
somme$cat[which(somme$somme<25)] <- "10-25"
somme$cat[which(somme$somme<10)] <- "<10"

ggplot(somme, aes(cat)) +
  geom_bar(fill = "steelblue")

df.multi.high <- df.multi
df.multi.high$geneId <- sapply(strsplit(rownames(df.multi.high), "\\.\\."), `[`, 1)
t <- as.data.frame(table(df.multi.high$geneId))
multi <- t[which(t$Freq>1),]

df.multi.high <- df.multi.high[df.multi.high$geneId %in% multi$Var1,]
df.multi.high <- df.multi.high[ , -which(names(df.multi.high) %in% c("geneId"))]

```

# MOB work on majoritary isoform

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

gg <- multi$Var1
majoritary <- character()
dat <- data.frame()
for(i in 1:length(gg)){
  idx <- grep(paste0("^",gg[i],"\\.\\."),rownames(df.multi.high))
  iso <- Matrix::rowSums(df.multi.high[idx,])
  x <- max(iso)/sum(iso)
  
  majoritary <- c(majoritary, names(which.max(iso)))
  
  de <- data.frame(gg[i],max(iso),sum(iso),x)
  names(de)<-c("gene","max","total","ratiomax")
  dat <- rbind(dat, de)
  
  #print(paste0(gg[i], ",", max(iso), ",", sum(iso), ",", x, ",", sep=""))
}

hs=hist(dat$ratiomax, breaks=100)
#str(hs)
#tail(cbind(hs$mids,hs$counts),20)
#barplot(hs$counts)
#summary(hs$counts)

ggplot(data=dat, aes(x=total,y=ratiomax)) +  
      geom_point(shape = 21, colour = "black", fill="lightsteelblue3") +
      theme_bw() + theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank()) + 
      scale_x_continuous(trans = 'log10')
      #geom_text_repel(data=subset(dat, dat$ratiomax < 0.5 & dat$total > 1000), aes(label=gene), size=5) +
      #geom_point(data=subset(dat, dat$ratiomax < 0.5 & dat$total > 1000), col="red") 

MOB[["MULTI"]] <- CreateAssayObject(counts = df.multi.high)
MOB <- NormalizeData(object = MOB, assay = "MULTI")
#CBS2 <- scale_my_data(CBS2, assay="MULTI")

```

# MOB isoform switch detection

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

DefaultAssay(object = MOB) <- "MULTI"
total <- data.frame()
totaladj <- data.frame()
for (i in (1:4)){
  k <- i+1
  for (j in (k:5)){
    if(i != j){
      print(paste(i, " ", j, " ",clusters[i], " vs ", clusters[j], sep=""))
      
      markers <- FindMarkers(object = MOB, ident.1=clusters[i], ident.2=clusters[j])
      markers$cluster <- clusters[j]
      markers$region.1 <- clusters[i]
      markers$region.2 <- clusters[j]
      #markers$contrast <- paste(clusters[i], "vs", clusters[j], sep=" ")
      markers[which(markers$avg_log2FC>0),]$cluster <- clusters[i]
      markers$geneId <- sapply(strsplit(rownames(markers), "\\.\\."), `[`, 1)
      markers$transcriptId <- sapply(strsplit(rownames(markers), "\\.\\."), `[`, 2)
      markers$majo <- 0
      markers$majo[which(rownames(markers) %in% majoritary)] <- 1
      markers <- subset(markers, markers$majo == 1 | (markers$p_val_adj < 0.05 & markers$majo == 0))
      all.genes <- unique(markers$geneId)
      for (k in (1:length(all.genes))){
         sub <- markers[which(markers$geneId == all.genes[k]),]
         nb.clusters <- unique(sub$cluster)
         nb.transcripts <- unique(sub$transcriptId)
        
         if(length(nb.clusters) > 1 & length(nb.transcripts) > 1){
           totaladj <- rbind(totaladj, sub)
         }
      }
      print (dim(totaladj))
      print (unique(totaladj$geneId))
    }
  }
}

write.table(totaladj, file="output/MOB.isoswitch.new.csv", sep=",")


```

# Session Info

```{r sessinf}
sessionInfo()
```

