---
title: "SiT - Figures"
author: '[Kevin Lebrigand](mailto:lebrigand@ipmc.cnrs.fr)§, Joseph Bergenstråhle §, Kim Thrane §, Annelie Mollbrink, Pascal Barbry, Rainer Waldmann and Joakim Lundeberg'
date: 'Compiled: `r format(Sys.Date(), "%B %d, %Y")`'
abstract: 'The spatial landscape of gene expression isoforms in tissue sections.'
output:
  html_document:
    df_print: paged
    toc: yes
    toc_depth: '3'
  html_notebook:
    code_folding: none
    theme: journal
    toc: yes
    toc_depth: 3
    toc_float: yes
  pdf_document:
    toc: yes
    toc_depth: '3'
---

<style type="text/css">

body, td {
   font-size: 15px;
}
code.r{
  font-size: 15px;
}
pre {
  font-size: 15px
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  cache = FALSE,
  cache.lazy = FALSE,
  tidy = TRUE
)

```

# Integration of external data (Tepe et al., Cell report, 2018) (GSE121891)

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

setwd("thisdir")
source("00-common.import.R")

raw_counts<-read.table(file="data/GSE121891_OB_6_runs_processed_seurat.dge.csv")
metadata<-read.table(file="data/GSE121891_OB_metaData_seurat.csv",sep=",", header=TRUE)
rownames(metadata) <- metadata$X
metadata <- metadata[,-c(1)]
metadata[1:5,1:7]
gse <- CreateSeuratObject(raw_counts, project = "GSE121891", assay = "RNA", meta.data = metadata)
gse <- subset(gse, subset = orig.ident %in% c("WT1","WT2"))
gse <- SCTransform(gse, ncells = 3000, verbose = FALSE) %>% RunPCA(verbose = FALSE) %>% RunUMAP(dims = 1:30)
tepe <- readRDS(tepe, "data/tepe.WT.rds")
DimPlot(tepe, group.by = "ClusterName", label=TRUE)

saveRDS(gse, "data/tepe.WT.rds")

```

# Session Info

```{r sessinf}
sessionInfo()
```

