---
title: "SiT - Figures"
author: '[Kevin Lebrigand](mailto:lebrigand@ipmc.cnrs.fr)§, Joseph Bergenstråhle §, Kim Thrane §, Annelie Mollbrink, Pascal Barbry, Rainer Waldmann and Joakim Lundeberg'
date: 'Compiled: `r format(Sys.Date(), "%B %d, %Y")`'
abstract: 'The spatial landscape of gene expression isoforms in tissue sections.'
output:
  html_document:
    df_print: paged
    toc: yes
    toc_depth: '3'
  html_notebook:
    code_folding: none
    theme: journal
    toc: yes
    toc_depth: 3
    toc_float: yes
  pdf_document:
    toc: yes
    toc_depth: '3'
---

<style type="text/css">

body, td {
   font-size: 15px;
}
code.r{
  font-size: 15px;
}
pre {
  font-size: 15px
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  cache = FALSE,
  cache.lazy = FALSE,
  tidy = TRUE
)

```


# Zeisel et al., 2018

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

setwd("thisdir")
source("00-common.import.R")

library(Seurat)
library(loomR)
library(hdf5r)
require(rlist)

lfile <- H5File$new("data/l5_all.loom")
full.matrix <- t(lfile[["matrix"]][,])
nrow(full.matrix)
ncol(full.matrix)
rownames(full.matrix) <- lfile[["row_attrs"]][["Gene"]][]
colnames(full.matrix) <- make.unique(lfile[["col_attrs"]][["CellID"]][], sep="_")
seurat <- CreateSeuratObject(full.matrix)
mito.genes <- grep(pattern = "^mt-", x = rownames(seurat@assays$RNA), value = TRUE)
dropouts <- Matrix::colSums(seurat@assays$RNA@data == 0)/nrow(seurat@assays$RNA)
ribo.genes <- grep(pattern = "^Rp[sl]", x = rownames(seurat@assays$RNA), value = TRUE)
percent.mito <- Matrix::colSums(seurat@assays$RNA[mito.genes, ])/Matrix::colSums(seurat@assays$RNA)
percent.ribo <- Matrix::colSums(seurat@assays$RNA[ribo.genes, ])/Matrix::colSums(seurat@assays$RNA)
seurat[['percent.mito']] <- percent.mito
seurat[['percent.ribo']] <- percent.ribo
seurat[['dropouts']] <- dropouts
fact <- c("ClusterName","OriginalClusters","Subclass","Region","TaxonomyRank1","TaxonomyRank2","TaxonomyRank3","TaxonomyRank4","Taxonomy_group")
for(i in 1:length(fact)){
  seurat[[fact[i]]] <- lfile[["col_attrs"]][[fact[i]]][]
}
seurat <- SetIdent(seurat, value="Taxonomy_group")

tsne <- data.frame(tSNE1=lfile[["col_attrs"]][["_tSNE1"]][], tSNE2=lfile[["col_attrs"]][["_tSNE2"]][])
colnames(tsne) <- c("tSNE_1","tSNE_2")
rownames(tsne) <- make.unique(lfile[["col_attrs"]][["CellID"]][], sep="_")
seurat[["tsne"]] <- CreateDimReducObject(embeddings = as.matrix(tsne), key = "tSNE_", assay = DefaultAssay(seurat))
seurat <- SetIdent(seurat, value="TaxonomyRank4")
DimPlot(seurat)

lfile$close_all()

x <- seq(from=1, to=length(rownames(seurat@meta.data)), by = 20)
length(x)
cellnames <- rownames(seurat@meta.data)[x]
sub <- subset(seurat, cells=cellnames)

sub <- SCTransform(object = sub, verbose = FALSE)
sub <- RunPCA(sub, verbose = FALSE)
sub <- RunUMAP(object = sub, dims = 1:40)
sub <- FindNeighbors(object = sub, do.plot=TRUE, dims = 1:40)
sub <- FindClusters(object = sub, resolution=0.2)

sub[['customClusters']] <- sub@active.ident
sub <- SetIdent(sub, value="Taxonomy_group")
DimPlot(object = sub, label=TRUE, reduction="umap")

saveRDS(sub,"data/zeisel.sub8k.rds")

```

# Session Info

```{r sessinf}
sessionInfo()
```

