---
title: "SiT - Mouse Coronal Brain Section (CBS1) - 03.switch"
author: '[Kevin Lebrigand](mailto:lebrigand@ipmc.cnrs.fr)§, Joseph Bergenstråhle §, Kim Thrane §, Annelie Mollbrink, Pascal Barbry, Rainer Waldmann and Joakim Lundeberg'
date: 'Compiled: `r format(Sys.Date(), "%B %d, %Y")`'
abstract: 'The spatial landscape of gene expression isoforms in tissue sections.'
output:
  html_document:
    df_print: paged
    toc: yes
    toc_depth: '3'
  html_notebook:
    code_folding: none
    theme: journal
    toc: yes
    toc_depth: 3
    toc_float: yes
  pdf_document:
    toc: yes
    toc_depth: '3'
---

<style type="text/css">

body, td {
   font-size: 15px;
}
code.r{
  font-size: 15px;
}
pre {
  font-size: 15px
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  cache = FALSE,
  cache.lazy = FALSE,
  tidy = TRUE
)

```

# Loading CBS1

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=4, fig.width=6}

setwd("/data/10x_data/10x_visium/")
source("00.import.R")

CBS1 <- readRDS("./output/CBS1.nanopore.rds")
clusters <- c("Midbrain","Hippocampus area","Isocortex-1","Isocortex-2","Olfactory area","Fiber tracts","Retrosplenial area","CA1/CA2","Thalamus","Hypothalamus","CA3","DG","Outside")

```

# Isoform switch detection

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

DefaultAssay(object = CBS1) <- "ISO"
total <- data.frame()
totaladj <- data.frame()
for (i in (1:12)){
  k <- i+1
  for (j in (k:13)){
    if(i != j){
      print(paste(i, " ", j, " ",clusters[i], " vs ", clusters[j], sep=""))
      
      markers <- FindMarkers(object = CBS1, ident.1=clusters[i], ident.2=clusters[j])
      markers$cluster <- clusters[j]
      markers$region.1 <- clusters[i]
      markers$region.2 <- clusters[j]
      #markers$contrast <- paste(clusters[i], "vs", clusters[j], sep=" ")
      markers[which(markers$avg_logFC>0),]$cluster <- clusters[i]
      markers$geneId <- sapply(strsplit(rownames(markers), "\\.\\."), `[`, 1)
      markers$transcriptId <- sapply(strsplit(rownames(markers), "\\.\\."), `[`, 2)
      markers <- markers[markers$p_val < 0.05,]
      all.genes <- unique(markers$geneId)
      for (k in (1:length(all.genes))){
         sub <- markers[which(markers$geneId == all.genes[k]),]
         nb.clusters <- unique(sub$cluster)
         nb.transcripts <- unique(sub$transcriptId)
        
         if(length(nb.clusters) > 1 & length(nb.transcripts) > 1){
           total <- rbind(total, sub)
         }
      }
      #print (dim(total))
      #print (length(unique(total$geneId)))
      markers <- markers[markers$p_val_adj < 0.05,]
      all.genes <- unique(markers$geneId)
      for (k in (1:length(all.genes))){
         sub <- markers[which(markers$geneId == all.genes[k]),]
         nb.clusters <- unique(sub$cluster)
         nb.transcripts <- unique(sub$transcriptId)
        
         if(length(nb.clusters) > 1 & length(nb.transcripts) > 1){
           totaladj <- rbind(totaladj, sub)
         }
      }
      #print (dim(total))
      #print (length(unique(total$geneId)))
    }
  }
}

```

# Top hits p_val_adj < 0.05

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

length(unique(totaladj$geneId))
print(unique(totaladj$geneId))
write.table(totaladj, file="output/CBS1.isoswitch.top.csv", sep=",")

```

# Lower hits p_val < 0.05

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

length(unique(total$geneId))
print(unique(total$geneId))
write.table(total, file="output/CBS1.isoswitch.csv", sep=",")

```

# Junctions usage switch detection

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=10}

DefaultAssay(object = CBS1) <- "JUNC"
total <- data.frame()
totaladj <- data.frame()
for (i in (1:12)){
  k <- i+1
  for (j in (k:13)){
    if(i != j){
      print(paste(i, " ", j, " ",clusters[i], " vs ", clusters[j], sep=""))
      
      markers <- FindMarkers(object = CBS1, ident.1=clusters[i], ident.2=clusters[j])
      markers$cluster <- clusters[j]
      markers$region.1 <- clusters[i]
      markers$region.2 <- clusters[j]
      #markers$contrast <- paste(clusters[i], "vs", clusters[j], sep=" ")
      markers[which(markers$avg_logFC>0),]$cluster <- clusters[i]
      markers$geneId <- sapply(strsplit(rownames(markers), "\\.\\."), `[`, 1)
      markers$junctionId <- sapply(strsplit(rownames(markers), "\\.\\."), `[`, 2)
      markers <- markers[markers$p_val < 0.05,]
      all.genes <- unique(markers$geneId)
      for (k in (1:length(all.genes))){
         sub <- markers[which(markers$geneId == all.genes[k]),]
         nb.clusters <- unique(sub$cluster)
         nb.junctions <- unique(sub$junctionId)
        
         if(length(nb.clusters) > 1 & length(nb.junctions) > 1){
           total <- rbind(total, sub)
         }
      }
      
      markers <- markers[markers$p_val_adj < 0.05,]
      all.genes <- unique(markers$geneId)
      for (k in (1:length(all.genes))){
         sub <- markers[which(markers$geneId == all.genes[k]),]
         nb.clusters <- unique(sub$cluster)
         nb.junctions <- unique(sub$junctionId)
        
         if(length(nb.clusters) > 1 & length(nb.junctions) > 1){
           totaladj <- rbind(totaladj, sub)
         }
      }
      
      print (dim(total))
      print (length(unique(total$geneId)))
    }
  }
}

```

# Top hits p_val_adj < 0.05

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

length(unique(totaladj$geneId))
print(unique(totaladj$geneId))
write.table(totaladj, file="output/CBS1.juncswitch.top.csv", sep=",")

```

# Lower hits p_val < 0.05

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

length(unique(total$geneId))
print(unique(total$geneId))
write.table(total, file="output/CBS1.juncswitch.csv", sep=",")

```

# Session Info

```{r sessinf}
sessionInfo()
```

