---
title: "SiT - Figures"
author: '[Kevin Lebrigand](mailto:lebrigand@ipmc.cnrs.fr)§, Joseph Bergenstråhle §, Kim Thrane §, Annelie Mollbrink, Pascal Barbry, Rainer Waldmann and Joakim Lundeberg'
date: 'Compiled: `r format(Sys.Date(), "%B %d, %Y")`'
abstract: 'The spatial landscape of gene expression isoforms in tissue sections.'
output:
  html_document:
    df_print: paged
    toc: yes
    toc_depth: '3'
  html_notebook:
    code_folding: none
    theme: journal
    toc: yes
    toc_depth: 3
    toc_float: yes
  pdf_document:
    toc: yes
    toc_depth: '3'
---

<style type="text/css">

body, td {
   font-size: 15px;
}
code.r{
  font-size: 15px;
}
pre {
  font-size: 15px
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  cache = FALSE,
  cache.lazy = FALSE,
  tidy = TRUE
)

```

```{r message=FALSE, warning=FALSE, echo=TRUE, eval=T, fig.height=4, fig.width=6}

setwd("/data/10x_data/10x_visium/000-github/")
source("00.common.import.R")
CBS1 <- readRDS("./output/CBS1.nanopore.rds")

```

# Deconvolution using Zeisel et al., 2018

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

zeisel <- readRDS("zeisel.sub8k.rds")
DimPlot(zeisel, label = TRUE)

#cluster_markers_all <- Seurat::FindAllMarkers(object = zeisel, 
#                                              assay = "SCT",
#                                              slot = "data",
#                                              verbose = TRUE, 
#                                              only.pos = TRUE)

#saveRDS(object = cluster_markers_all, file = here::here("markers_zeisel_TaxonomyRank4.rds"))

#cluster_markers_all <- readRDS("markers_zeisel_TaxonomyRank4.rds")

#set.seed(123)
#spotlight_ls <- spotlight_deconvolution(
#  se_sc = zeisel,
#  counts_spatial = CBS1@assays$Spatial@counts,
#  clust_vr = "TaxonomyRank4", # Variable in sc_seu containing the cell-type annotation
#  cluster_markers = cluster_markers_all, # Dataframe with the marker genes
#  cl_n = 100, # number of cells per cell type to use
#  hvg = 3000, # Number of HVG to use
#  ntop = NULL, # How many of the marker genes to use (by default all)
#  transf = "uv", # Perform unit-variance scaling per cell and spot prior to factorzation and NLS
#  method = "nsNMF", # Factorization method
#  min_cont = 0 # Remove those cells contributing to a spot below a certain threshold 
#)
#saveRDS(object = spotlight_ls, file = here::here("spotlight_zeisel_TaxonomyRank4_CBS1.rds"))

spotlight_ls <- readRDS("spotlight_zeisel_TaxonomyRank4_CBS1.rds")

nmf_mod <- spotlight_ls[[1]]
decon_mtrx <- spotlight_ls[[2]]

h <- NMF::coef(nmf_mod[[1]])
rownames(h) <- paste("Topic", 1:nrow(h), sep = "_")
topic_profile_plts <- SPOTlight::dot_plot_profiles_fun(
  h = h,
  train_cell_clust = nmf_mod[[2]])

topic_profile_plts[[2]] + ggplot2::theme(
  axis.text.x = ggplot2::element_text(angle = 90), 
  axis.text = ggplot2::element_text(size = 12))

topic_profile_plts[[1]] + theme(axis.text.x = element_text(angle = 90), 
                                axis.text = element_text(size = 12))

basis_spotlight <- data.frame(NMF::basis(nmf_mod[[1]]))

colnames(basis_spotlight) <- unique(stringr::str_wrap(nmf_mod[[2]], width = 30))

#######################################################
# do not remove contributing cell types under 8%
# we keep all contributions for correlation computation
# this is usefull for scatterpie spotligight plot
# not to have too many contributing cells
#######################################################
#decon_mtrx_sub <- decon_mtrx[, colnames(decon_mtrx) != "res_ss"]
#decon_mtrx_sub[decon_mtrx_sub < 0.08] <- 0
#decon_mtrx <- cbind(decon_mtrx_sub, "res_ss" = decon_mtrx[, "res_ss"])
rownames(decon_mtrx) <- colnames(CBS1)

decon_df <- decon_mtrx %>%
  data.frame() %>%
  tibble::rownames_to_column("barcodes")

CBS1@meta.data <- CBS1@meta.data %>%
  tibble::rownames_to_column("barcodes") %>%
  dplyr::left_join(decon_df, by = "barcodes") %>%
  tibble::column_to_rownames("barcodes")


#saveRDS(CBS1, "./output/CBS1.nanopore.deconv.rds")

#Seurat::SpatialFeaturePlot(
#  object = CBS1,
#  features = c("Oligodendrocytes"),
#  alpha = c(0.1, 1))

cell_types_all <- colnames(decon_mtrx)[which(colnames(decon_mtrx) != "res_ss")]

#pdf("spotlight.CBS1.pdf", width=16, height=12, useDingbats=FALSE)
SPOTlight::spatial_scatterpie(se_obj = CBS1,
                              cell_types_all = cell_types_all,
                              img_path = "/data/10x_data/10x_visium/illumina/CBS1/spatial/tissue_lowres_image.png",
                              pie_scale = 0.4)
#dev.off()

#pdf("spotlight.CBS1.pdf", width=16, height=12, useDingbats=FALSE)
#SPOTlight::spatial_scatterpie(se_obj = CBS1,
#                              cell_types_all = c("Oligodendrocytes","Microglia"),
#                              img_path = "/data/10x_data/10x_visium/illumina/CBS1/spatial/tissue_lowres_image.png",
#                              pie_scale = 0.4)
#dev.off()

int <- c("Oligodendrocytes","Peptidergic.neurons","Dentate.gyrus.granule.neurons","Dentate.gyrus.radial.glia.like.cells","Di..and.mesencephalon.excitatory.neurons","Subventricular.zone.radial.glia.like.cells","Telencephalon.projecting.excitatory.neurons")

#pdf("spotlight.CBS1.alltype.pdf", width=32, height=8, useDingbats=FALSE)
#Seurat::SpatialFeaturePlot(
#  object = CBS1,
#  features = int,
#  alpha = c(0.1, 1))
#dev.off()


#SPOTlight::spatial_scatterpie(se_obj = CBS1,
#                              cell_types_all = c("MyOligo","OPC","OEC1","OEC2","OEC3","OEC4","OEC5","N2"),
#                              img_path = "/data/10x_data/10x_visium/illumina/CBS1/spatial/tissue_lowres_image.png",
#                              pie_scale = 0.6)

```

# Session Info

```{r sessinf}
sessionInfo()
```

