---
title: "SiT - Mouse Coronal Brain Section (CBS2) - 02.nanopore"
author: '[Kevin Lebrigand](mailto:lebrigand@ipmc.cnrs.fr)§, Joseph Bergenstråhle §, Kim Thrane §, Annelie Mollbrink, Pascal Barbry, Rainer Waldmann and Joakim Lundeberg'
date: 'Compiled: `r format(Sys.Date(), "%B %d, %Y")`'
abstract: 'The spatial landscape of gene expression isoforms in tissue sections.'
output:
  html_document:
    df_print: paged
    toc: yes
    toc_depth: '3'
  html_notebook:
    code_folding: none
    theme: journal
    toc: yes
    toc_depth: 3
    toc_float: yes
  pdf_document:
    toc: yes
    toc_depth: '3'
---

<style type="text/css">

body, td {
   font-size: 15px;
}
code.r{
  font-size: 15px;
}
pre {
  font-size: 15px
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  cache = FALSE,
  cache.lazy = FALSE,
  tidy = TRUE
)

```

# Loading Spatial data

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=4, fig.width=6}

setwd("/data/10x_data/10x_visium/")
source("00.import.R")

CBS2 <- readRDS("./output/CBS2.rds")

```

# Add Nanopore data: ISO, ISOG, JUNC and AtoI assays, scale_my_data

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

all = read.delim("./nanopore/CBS2_genematrix.txt", stringsAsFactors = F)
data = all[,2:ncol(all)]
colnames(data) <- paste(colnames(data),"-1", sep="")
rownames(data) <- all$geneId
vect <- colnames(CBS2@assays$Spatial)
data <- data[vect]
CBS2[["ISOG"]] <- CreateAssayObject(counts = data)
CBS2 <- NormalizeData(object = CBS2, assay = "ISOG")
CBS2 <- ScaleData(object = CBS2, assay = "ISOG")

all = read.delim("./nanopore/CBS2_isomatrix.txt", stringsAsFactors = F)
data = all[,4:ncol(all)]
dim(data)
colnames(data) <- paste(colnames(data),"-1", sep="")
rownames(data) = paste(all$geneId, all$transcriptId, sep="..")
vect <- colnames(CBS2@assays$Spatial)
idx <- grep("undef", rownames(data), invert = TRUE)
data <- data[idx,vect]
dim(data)
CBS2[["ISO"]] <- CreateAssayObject(counts = data)
CBS2 <- NormalizeData(object = CBS2, assay = "ISO")
CBS2 <- scale_my_data(CBS2, assay="ISO")

all = read.delim("./nanopore/CBS2_juncmatrix.txt", stringsAsFactors = F)
data = all[,2:ncol(all)]
dim(data)
colnames(data) <- paste(colnames(data),"-1", sep="")
rownames(data) = str_replace(all$junctionId, ":", "..")
vect <- colnames(CBS2@assays$Spatial)
data <- data[,vect]
data <- data[which(Matrix::rowSums(data)>9),]
dim(data)
CBS2[["JUNC"]] <- CreateAssayObject(counts = data)
CBS2 <- NormalizeData(object = CBS2, assay = "JUNC")
CBS2 <- scale_my_data(CBS2, assay="JUNC")

all = read.delim("./nanopore/CBS2.snp100838.RN3.QV6_snpmatrix.csv", sep=",", stringsAsFactors = F)
data = all[,4:ncol(all)]
dim(data)
colnames(data) <- paste(colnames(data),"-1", sep="")
rownames(data) = paste(all$geneId, all$transcriptId, sep=":")
vect <- colnames(CBS2@assays$Spatial)
#idx <- grep("intergenic", rownames(data), invert = TRUE)
#data <- data[idx,vect]
#idx <- grep("Rik", rownames(data), invert = TRUE)
data <- data[,vect]
idx1 <- grep("\\.\\.A", rownames(data))
idx2 <- grep("\\.\\.G", rownames(data))
data <- data[c(idx1,idx2),]
dim(data)

x <- unique(sapply(strsplit(rownames(data), "\\.\\."), `[`, 1))
countA <- data[grep("\\.\\.A",rownames(data)),]
colnames(countA) <- colnames(data)
rownames(countA) <- x
countG <- data[grep("\\.\\.G",rownames(data)),]
colnames(countG) <- colnames(data)
rownames(countG) <- x
ratio <- countG/(countA+countG)

#countA[is.na(countA)] <- 0
#countG[is.na(countG)] <- 0

CBS2[['AtoI_A']] <- Matrix::colSums(countA)
CBS2[['AtoI_G']] <- Matrix::colSums(countG)
CBS2[['AtoI_total']] <- CBS2[['AtoI_A']]+CBS2[['AtoI_G']]
CBS2[['AtoI_ratio']] <- CBS2[['AtoI_G']]/CBS2[['AtoI_total']]
CBS2[['AtoI_detected']] <- Matrix::colSums(countG > 0)

CBS2[["AtoI"]] <- CreateAssayObject(counts = countA)
CBS2 <- SetAssayData(CBS2, slot = "data", as.matrix(countG), assay = "AtoI")
CBS2 <- SetAssayData(CBS2, slot = "scale.data", as.matrix(ratio), assay = "AtoI")

saveRDS(CBS2, "./output/CBS2.nanopore.rds")


```

# Session Info

```{r sessinf}
sessionInfo()
```

