---
title: "SiT - Figures"
author: '[Kevin Lebrigand](mailto:lebrigand@ipmc.cnrs.fr)§, Joseph Bergenstråhle §, Kim Thrane §, Annelie Mollbrink, Pascal Barbry, Rainer Waldmann and Joakim Lundeberg'
date: 'Compiled: `r format(Sys.Date(), "%B %d, %Y")`'
abstract: 'The spatial landscape of gene expression isoforms in tissue sections.'
output:
  html_document:
    df_print: paged
    toc: yes
    toc_depth: '3'
  html_notebook:
    code_folding: none
    theme: journal
    toc: yes
    toc_depth: 3
    toc_float: yes
  pdf_document:
    toc: yes
    toc_depth: '3'
---

<style type="text/css">

body, td {
   font-size: 15px;
}
code.r{
  font-size: 15px;
}
pre {
  font-size: 15px
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  cache = FALSE,
  cache.lazy = FALSE,
  tidy = TRUE
)

```

```{r message=FALSE, warning=FALSE, echo=TRUE, eval=T, fig.height=4, fig.width=6}

setwd("thisdir")
source("00-common.import.R")

CBS2 <- readRDS("output/CBS2.nanopore.deconv.rds")

DefaultAssay(CBS2) <- "SCT"
CBS2 <- subset(CBS2, subset = nFeature_ISO>1)

pdf("figures/Figure.sup.07.pdf", width=8, height=8, useDingbats=FALSE)
SpatialDimPlot(CBS2, pt.size.factor =1.6, label=TRUE)
SpatialFeaturePlot(object = CBS2, features = c("C1ql2","Myo5b","Slc6a3","Zic1","Fibcd1","Nrep","Mag","Hpcal1","Camk2n1","Myl4","Gfap","Mbp"), pt.size.factor=1.6, alpha = c(1, 1))
DotPlot(object = CBS2, features = rev(c("C1ql2","Myo5b","Slc6a3","Zic1","Fibcd1","Nrep","Mag","Hpcal1","Camk2n1","Myl4","Gfap","Mbp")), scale=TRUE, scale.max = 100, scale.min = 0, col.min = -1, col.max = 2, cols = c("lightgrey", "red")) + RotatedAxis()

dev.off()

```

# Session Info

```{r sessinf}
sessionInfo()
```

