---
title: "SiT - Mouse Coronal Brain Section (CBS2) - 01.spatial"
author: '[Kevin Lebrigand](mailto:lebrigand@ipmc.cnrs.fr)§, Joseph Bergenstråhle §, Kim Thrane §, Annelie Mollbrink, Pascal Barbry, Rainer Waldmann and Joakim Lundeberg'
date: 'Compiled: `r format(Sys.Date(), "%B %d, %Y")`'
abstract: 'The spatial landscape of gene expression isoforms in tissue sections.'
output:
  pdf_document:
    toc: yes
    toc_depth: '3'
  html_notebook:
    code_folding: none
    theme: journal
    toc: yes
    toc_depth: 3
    toc_float: yes
  html_document:
    df_print: paged
    toc: yes
    toc_depth: '3'
---

<style type="text/css">

body, td {
   font-size: 15px;
}
code.r{
  font-size: 15px;
}
pre {
  font-size: 15px
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  cache = FALSE,
  cache.lazy = FALSE,
  tidy = TRUE
)
```

# Loading Spatial data

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=4, fig.width=6}

setwd("/data/10x_data/10x_visium/000-github")
source("00.common.import.R")
clusters <- c("Midbrain","Hippocampus area","Isocortex-1","Isocortex-2","Olfactory area","Fiber tracts","Retrosplenial area","CA1/CA2","Thalamus","Hypothalamus","CA3","DG")

CBS1 <- readRDS("./output/CBS1.nanopore.deconv.rds")
CBS2 <- readRDS("./output/CBS2.nanopore.deconv.rds")

p1 <- SpatialDimPlot(CBS2)
p2 <- SpatialDimPlot(CBS1)

CBS1 <- subset(CBS1,ident="Outside",invert=TRUE)
pdf("Fig.3a.pdf", width=12, height=8, useDingbats=FALSE)
plot_grid(p1,p2)
dev.off()

```

# Multi-isoforms genes

```{r message=FALSE, warning=FALSE, echo=TRUE, eval=T, fig.height=4, fig.width=6}

dim(CBS2@assays$ISO@counts)
df <- as.data.frame(Matrix::rowSums(CBS2@assays$ISO@counts))
df$gene <- sapply(strsplit(rownames(df), "\\.\\."), `[`, 1)
df$ensembl_transcript_id <- sapply(strsplit(sapply(strsplit(rownames(df), "\\.\\."), `[`, 2), "\\."), `[`, 1)
head(df)
length(unique(df$gene))

xx <- as.data.frame(table(df$gene))
colnames(xx) <- c("gene","nb.iso")
table(xx$nb.iso)

library(biomaRt)
ensembl = useMart("ensembl")
ensembl = useDataset("mmusculus_gene_ensembl", mart = ensembl)
lengths = getBM(attributes=c("ensembl_gene_id", "ensembl_transcript_id", "transcript_length", "transcript_biotype", "external_gene_name"), values = df[,"ensembl_transcript_id"], mart = ensembl)
dd <- merge(df, lengths)
dd2 <- merge(dd,xx)

give.n <- function(x){
  return(c(y = median(x)*1.05, label = median(x))) 
  # experiment with the multiplier to find the perfect position
}

p1 <- ggplot(data=dd2, aes(x=transcript_biotype, y=transcript_length, group=transcript_biotype))+
  geom_boxplot(outlier.shape = NA, color="black") + theme_minimal() + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  stat_summary(fun.data = give.n, geom = "text", size=3, position = position_dodge(width = 0.5)) + 
  ylim(0, 5000)

p2 <- ggplot(data=dd2, aes(x=nb.iso, y=transcript_length, group=nb.iso))+
  geom_boxplot(outlier.shape = NA, color="black") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  stat_summary(fun.data = give.n, geom = "text", size=3, position = position_dodge(width = 1)) + 
  ylim(0, 5000) + theme_minimal()

plot_grid(p1,p2)

dim(CBS1@assays$ISO@counts)
df <- as.data.frame(Matrix::rowSums(CBS1@assays$ISO@counts))
df$gene <- sapply(strsplit(rownames(df), "\\.\\."), `[`, 1)
df$ensembl_transcript_id <- sapply(strsplit(sapply(strsplit(rownames(df), "\\.\\."), `[`, 2), "\\."), `[`, 1)
head(df)
length(unique(df$gene))

xx <- as.data.frame(table(df$gene))
colnames(xx) <- c("gene","nb.iso")
table(xx$nb.iso)

library(biomaRt)
ensembl = useMart("ensembl")
ensembl = useDataset("mmusculus_gene_ensembl", mart = ensembl)
lengths = getBM(attributes=c("ensembl_gene_id", "ensembl_transcript_id", "transcript_length", "transcript_biotype", "external_gene_name"), values = df[,"ensembl_transcript_id"], mart = ensembl)
dd <- merge(df, lengths)
dd2 <- merge(dd,xx)

give.n <- function(x){
  return(c(y = median(x)*1.05, label = median(x))) 
  # experiment with the multiplier to find the perfect position
}

p1 <- ggplot(data=dd2, aes(x=transcript_biotype, y=transcript_length, group=transcript_biotype))+
  geom_boxplot(outlier.shape = NA, color="black") + theme_minimal() + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  stat_summary(fun.data = give.n, geom = "text", size=3, position = position_dodge(width = 0.5)) + 
  ylim(0, 5000)

p2 <- ggplot(data=dd2, aes(x=nb.iso, y=transcript_length, group=nb.iso))+
  geom_boxplot(outlier.shape = NA, color="black") + theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) +
  stat_summary(fun.data = give.n, geom = "text", size=3, position = position_dodge(width = 1)) + 
  ylim(0, 5000) + theme_minimal()

plot_grid(p1,p2)

```

# Isoforms and genes number per region

```{r message=FALSE, warning=FALSE, echo=TRUE, eval=T, fig.height=4, fig.width=6}

ggplot(CBS2@meta.data, aes(x=ClusterName, y=nFeature_ISO, fill=ClusterName)) + geom_boxplot(outlier.shape = NA)+ ggtitle("Isoforms per region") + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

df <- data.frame()
for (i in (1:length(clusters))){
  
    print(paste0(clusters[i], dim(df)))
  
    sub <- subset(CBS2, ident=clusters[i])
    x <- data.frame(Matrix::rowSums(sub@assays$ISO@counts))
    x$delete <- 1
    colnames(x)<-c("count","delete")
    x <- x[x$count > 0,]
    z <- sub@assays$ISO@counts[rownames(x),]
    for (j in (1:length(colnames(z)))){
      dd <- z[,j]
      dd <- dd[dd != 0]
      nb.iso <- length(dd)
      nb.gene  <- length(unique(sapply(strsplit(names(dd), "\\.\\."), `[`, 1)))
      df2 <- data.frame(colnames(z)[j],nb.iso,nb.gene,clusters[i])
      colnames(df2)<-c("id","isoform","gene","region")
      df <- rbind(df, df2)
      
    }
}

df$region <- factor(df$region, levels = c("Midbrain","Hippocampus area","Isocortex-1","Isocortex-2","Olfactory area","Fiber tracts","Retrosplenial area","CA1/CA2","Thalamus","Hypothalamus","CA3","DG"))

ggplot(df, aes(x=region, y=isoform, fill=region)) + geom_boxplot(outlier.shape = NA)+ ggtitle("Isoforms per region") + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))
ggplot(df, aes(x=region, y=gene, fill=region)) + geom_boxplot(outlier.shape = NA)+ ggtitle("Genes per region") + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

p1 <- ggplot(df, aes(x=region, y=isoform/gene, fill=region)) + geom_boxplot(outlier.shape = NA)+ ggtitle("Isoforms/Genes per region CBS2") + theme_minimal() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

# CBS1 
df.cbs1 <- data.frame()
for (i in (1:length(clusters))){
  
    print(paste0(clusters[i], dim(df.cbs1)))
  
    sub <- subset(CBS1, ident=clusters[i])
    x <- data.frame(Matrix::rowSums(sub@assays$ISO@counts))
    x$delete <- 1
    colnames(x)<-c("count","delete")
    x <- x[x$count > 0,]
    z <- sub@assays$ISO@counts[rownames(x),]
    for (j in (1:length(colnames(z)))){
      dd <- z[,j]
      dd <- dd[dd != 0]
      nb.iso <- length(dd)
      nb.gene  <- length(unique(sapply(strsplit(names(dd), "\\.\\."), `[`, 1)))
      df2 <- data.frame(colnames(z)[j],nb.iso,nb.gene,clusters[i])
      colnames(df2)<-c("id","isoform","gene","region")
      df.cbs1 <- rbind(df.cbs1, df2)
      
    }
}
df.cbs1$region <- factor(df.cbs1$region, levels = c("Midbrain","Hippocampus area","Isocortex-1","Isocortex-2","Olfactory area","Fiber tracts","Retrosplenial area","CA1/CA2","Thalamus","Hypothalamus","CA3","DG"))

p2 <- ggplot(df.cbs1, aes(x=region, y=isoform/gene, fill=region)) + geom_boxplot(outlier.shape = NA)+ ggtitle("Isoforms/Genes per region CBS1") + theme_minimal() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

plot_grid(p2,p1)

pdf("Fig.3d.pdf", width=12, height=6, useDingbats=FALSE)
plot_grid(p2,p1)
dev.off()

```

# Spot-2-Spot direct correlation

```{r message=FALSE, warning=FALSE, echo=TRUE, eval=T, fig.height=4, fig.width=6}

samples <- c("/data/10x_data/10x_visium/illumina/CBS1/filtered_feature_bc_matrix.h5","/data/10x_data/10x_visium/illumina/CBS2/filtered_feature_bc_matrix.h5")
imgs <- c("/data/10x_data/10x_visium/illumina/CBS1/spatial/tissue_hires_image.png","/data/10x_data/10x_visium/illumina/CBS2/spatial/tissue_hires_image.png")
spotfiles <- c("/data/10x_data/10x_visium/illumina/CBS1/spatial/tissue_positions_list.csv","/data/10x_data/10x_visium/illumina/CBS2/spatial/tissue_positions_list.csv")
json <- c("/data/10x_data/10x_visium/illumina/CBS1/spatial/scalefactors_json.json","/data/10x_data/10x_visium/illumina/CBS2/spatial/scalefactors_json.json")
infoTable <- data.frame(samples, imgs, spotfiles, json, stringsAsFactors = FALSE)
#saveRDS(object = infoTable, file = "CBS2_infoTable_filtered")

se <- InputFromTable(infotable = infoTable, 
                      min.gene.count = 0, 
                      min.gene.spots = 0,
                      min.spot.count = 0,
                      platform =  "Visium")

se <- LoadImages(se, time.resolve = F, verbose = T)
se <- MaskImages(object = se, compactness = 2)
ImagePlot(se, ncols = 2, method = "raster", type = "masked", darken = F) # Masked image
se <- AlignImages(se)

df <- as.data.frame(CBS1@meta.data$ClusterName)
rownames(df) <- paste0(rownames(CBS1@meta.data),"_1",sep="")
colnames(df) <- "ClusterName"
df2 <- as.data.frame(CBS2@meta.data$ClusterName)
rownames(df2) <- paste0(rownames(CBS2@meta.data),"_2",sep="")
colnames(df2) <- "ClusterName"
dd <- rbind(df,df2)
se[['ClusterName']] <- dd

se <- MaskImages(object = se)
FeatureOverlay(se, features = "ClusterName", sampleids = 2, ncols.samples = 2, dark.theme = F)
p1 <- SpatialDimPlot(CBS2)
p2 <- SpatialDimPlot(CBS1)

CBS1 <- subset(CBS1,ident="Outside",invert=TRUE)
pdf("SpatialDimPlot.pdf", width=12, height=8, useDingbats=FALSE)
plot_grid(p1,p2)
dev.off()

#pdf("out.pdf", width=12, height=6, useDingbats=FALSE)
#FeatureOverlay(se, features = "Nrgn", 
#                sampleids = 1:2,
#                cols = c("dark blue", "cyan", "yellow", "red", "dark red"),
#                method = "raster", ncols.samples = 2,  pt.size = 1.8)
#dev.off()

library(raster)

st.object <- GetStaffli(se)
x <- data.frame(rownames(st.object@meta.data[which(st.object@meta.data$sample==2),]),st.object@meta.data[which(st.object@meta.data$sample==2),]$warped_x, st.object@meta.data[which(st.object@meta.data$sample==2),]$warped_y)
y <- data.frame(rownames(st.object@meta.data[which(st.object@meta.data$sample==1),]),st.object@meta.data[which(st.object@meta.data$sample==1),]$original_x, st.object@meta.data[which(st.object@meta.data$sample==1),]$original_y)
colnames(x) <- c("ID","HIGH_PRCN_LAT","HIGH_PRCN_LON")
colnames(y) <- c("SRC_ID","HIGH_PRCN_LAT","HIGH_PRCN_LON")

d <- pointDistance(x[,3:2], y[,3:2], lonlat=FALSE, allpairs=T) 
i <- apply(d, 1, which.min)

x$SRC_ID = y$SRC_ID[i]
x$distance = d[cbind(1:nrow(d), i)]
x

#pdf("spot2spot.pdf", width=6, height=6, useDingbats=FALSE)
par(bg = 'white')
plot(x[,3:2], col="#00478E", pch=20, cex = .7)
points(y[,3:2], col="#2B8CC7", pch=20, cex = .7)
for (i in 1:nrow(x)) {
    j <- y$SRC_ID==x$SRC_ID[i]
    arrows(x[i,3], x[i,2], y[j,3], y[j,2],length=.05, col="grey") 
}
#dev.off()

x$corr.spatial <- 0
x$corr.isog <- 0
x$corr.iso <- 0

# either load the correlation
#x2 <- read.csv(file="spot2spot.correlation.last.csv", sep=",")
# and plot which direct spot is in the same cluster
#x2$color <- "red"
#x2$color[x2$SAME == TRUE] <- "green"
#table(x2$color)
#plot(x2[,3:2], col=x2$color, pch=20, cex = 1)

# or compute it directly
df1.spatial <- as.data.frame(CBS1@assays$Spatial@counts)
colnames(df1.spatial) <- paste0(colnames(df1.spatial),"_1")
df2.spatial <- as.data.frame(CBS2@assays$Spatial@counts)
colnames(df2.spatial) <- paste0(colnames(df2.spatial),"_2")
spatial <- merge(df1.spatial, df2.spatial, by=0, all = TRUE)
spatial[is.na(spatial)] <- 0
rownames(spatial) <- spatial$Row.names
spatial <- spatial[,-c(1)]

df1.isog <- as.data.frame(CBS1@assays$ISOG@counts)
colnames(df1.isog) <- paste0(colnames(df1.isog),"_1")
df2.isog <- as.data.frame(CBS2@assays$ISOG@counts)
colnames(df2.isog) <- paste0(colnames(df2.isog),"_2")
isog <- merge(df1.isog, df2.isog, by=0, all = TRUE)
isog[is.na(isog)] <- 0
rownames(isog) <- isog$Row.names
isog <- isog[,-c(1)]

df1.iso <- as.data.frame(CBS1@assays$ISO@counts)
colnames(df1.iso) <- paste0(colnames(df1.iso),"_1")
df2.iso <- as.data.frame(CBS2@assays$ISO@counts)
colnames(df2.iso) <- paste0(colnames(df2.iso),"_2")
iso <- merge(df1.iso, df2.iso, by=0, all = TRUE)
iso[is.na(iso)] <- 0
rownames(iso) <- iso$Row.names
iso <- iso[,-c(1)]

for (i in (1:dim(x)[1])){
#for (i in (1:10)){
  
  cell1 <- x[i,]$SRC_ID
  cell2 <- x[i,]$ID
  
  # Spatial
  df <- spatial[,c(cell1,cell2)]
  df <- df[which(df[,cell1]>0 | df[,cell2]>0),]
  x[i,]$corr.spatial <- cor.test(df[,cell1], df[,cell2])$estimate
  
  # ISOG
  df <- isog[,c(cell1,cell2)]
  df <- df[which(df[,cell1]>0 | df[,cell2]>0),]
  x[i,]$corr.isog <- cor.test(df[,cell1], df[,cell2])$estimate
  
  # ISO
  df <- iso[,c(cell1,cell2)]
  df <- df[which(df[,cell1]>0 | df[,cell2]>0),]
  x[i,]$corr.iso <- cor.test(df[,cell1], df[,cell2])$estimate

  print(x[i,])
}

metadata <- x[,c("ID","SAME","corr.spatial","corr.isog","corr.iso")]
rownames(metadata) <- gsub("_2", "",x$ID)
del <- metadata$SAME
names(del) <- rownames(metadata)
CBS2[['SAME']] <- del
del <- metadata$corr.spatial
names(del) <- rownames(metadata)
CBS2[['corr']] <- del
del <- metadata$corr.isog
names(del) <- rownames(metadata)
CBS2[['corr.isog']] <- del
del <- metadata$corr.iso
names(del) <- rownames(metadata)
CBS2[['corr.iso']] <- del

pdf("same.pdf", width=8, height=6, useDingbats=FALSE)
SpatialFeaturePlot(CBS2, c("SAME"))
dev.off()

pdf("corr.pdf", width=14, height=6, useDingbats=FALSE)
SpatialFeaturePlot(CBS2, c("corr","corr.isog","corr.iso"), min.cutoff = 0.5)
dev.off()

x$CBS2.ident <- CBS2@active.ident[gsub("_2", "", x$ID)]
x$CBS1.ident <- CBS1@active.ident[gsub("_1", "", x$SRC_ID)]
x$SAME <- ifelse(as.character(x$CBS2.ident)==as.character(x$CBS1.ident),TRUE,FALSE)
table(x$SAME)
write.table(x, file="spot2spot.correlation.last.csv", sep=",")

median(x$corr.spatial)
median(x$corr.isog)
median(x$corr.iso)

df.spatial <- data.frame(x$corr.spatial,"spatial",x$SAME)
colnames(df.spatial) <- c("cor","type","same")
df.isog <- data.frame(x$corr.isog,"isog",x$SAME)
colnames(df.isog) <- c("cor","type","same")
df.iso <- data.frame(x$corr.iso,"iso",x$SAME)
colnames(df.iso) <- c("cor","type","same")

whole <- rbind(df.spatial,df.isog,df.iso)
head(whole)

pdf("spot2spot.corr.pdf", width=8, height=6, useDingbats=FALSE)
ggplot(whole, aes(x=type, y=cor, fill=type)) +
  geom_boxplot(alpha=0.3)+
  ggtitle("Spot-2-spot transcriptome correlation") +
  scale_fill_brewer(palette="Dark2") +
  theme_classic()
dev.off()


pdf("sdp.CBS2.pdf", width=8, height=6, useDingbats=FALSE)
SpatialDimPlot(CBS2, pt.size.factor = 1.6) + ggtitle("CBS2")
dev.off()
sub <- subset(CBS1, ident="Outside", invert=TRUE)
pdf("sdp.CBS1.pdf", width=8, height=6, useDingbats=FALSE)
SpatialDimPlot(sub, pt.size.factor = 1.6) + ggtitle("CBS1")
dev.off()


```

# Gene-2-gene bulk correlation CBS1/CBS2

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=10}

df <- Matrix::rowSums(CBS1@assays$Spatial@counts)
df2 <- Matrix::rowSums(CBS2@assays$Spatial@counts)
de <- merge(df, df2, by=0, all=TRUE)
de[is.na(de)] <- 0
rownames(de)<- de$Row.names
de <- de[,-c(1)]
colnames(de) <- c("CBS1","CBS2")
head(de)
x <- cor.test(de$CBS1, de$CBS2, method="pearson")$estimate
x
x*x
dim(de)

#de <- de[de$CBS1>0 & de$CBS2>0,]
#x <- cor.test(de$CBS1, de$CBS2, method="pearson")$estimate
#x
de$ratio <- log(de$CBS2+1,2)/log(de$CBS1+1,2)
de$id <- rownames(de)
options(scipen=5)
p1 <- ggplot(data=de, aes(x=CBS1,y=CBS2)) +  
      geom_point(shape = 21, colour = "black", fill="lightsteelblue3") +
      theme_bw() + theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank()) + 
      scale_x_continuous(trans = 'log10') + scale_y_continuous(trans = 'log10')
      geom_text_repel(data=subset(de, de$ratio > 1.3 & de$CBS2 > 500), aes(label=id), size=3) +
      geom_point(data=subset(de, de$ratio > 1.3 & de$CBS2 > 500), col="red") +
      geom_text_repel(data=subset(de, de$ratio < 0.5 & de$CBS1 > 500), aes(label=id), size=3) +
      geom_point(data=subset(de, de$ratio < 0.5 & de$CBS1 > 500), col="red")

df <- Matrix::rowSums(CBS1@assays$ISOG@counts)
df2 <- Matrix::rowSums(CBS2@assays$ISOG@counts)
de <- merge(df, df2, by=0, all=TRUE)
de[is.na(de)] <- 0
rownames(de)<- de$Row.names
de <- de[,-c(1)]
colnames(de) <- c("CBS1","CBS2")
head(de)
x <- cor.test(de$CBS1, de$CBS2, method="pearson")$estimate
x
x*x
dim(de)

#de <- de[de$CBS1>0 & de$CBS2>0,]
#x <- cor.test(de$CBS1, de$CBS2, method="pearson")$estimate
#x
de$ratio <- log(de$CBS2+1,2)/log(de$CBS1+1,2)
de$id <- rownames(de)
options(scipen=5)
p2 <- ggplot(data=de, aes(x=CBS1,y=CBS2)) +  
      geom_point(shape = 21, colour = "black", fill="lightsteelblue3") +
      theme_bw() + theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank()) + 
      scale_x_continuous(trans = 'log10') + scale_y_continuous(trans = 'log10')
      geom_text_repel(data=subset(de, de$ratio > 1.3 & de$CBS2 > 500), aes(label=id), size=3) +
      geom_point(data=subset(de, de$ratio > 1.3 & de$CBS2 > 500), col="red") +
      geom_text_repel(data=subset(de, de$ratio < 0.5 & de$CBS1 > 500), aes(label=id), size=3) +
      geom_point(data=subset(de, de$ratio < 0.5 & de$CBS1 > 500), col="red")

df <- Matrix::rowSums(CBS1@assays$ISO@counts)
df2 <- Matrix::rowSums(CBS2@assays$ISO@counts)
de <- merge(df, df2, by=0, all=TRUE)
de[is.na(de)] <- 0
rownames(de)<- de$Row.names
de <- de[,-c(1)]
colnames(de) <- c("CBS1","CBS2")
head(de)
x <- cor.test(de$CBS1, de$CBS2, method="pearson")$estimate
x
x*x
dim(de)

#de <- de[de$CBS1>0 & de$CBS2>0,]
#x <- cor.test(de$CBS1, de$CBS2, method="pearson")$estimate
#x
de$ratio <- log(de$CBS2+1,2)/log(de$CBS1+1,2)
de$id <- rownames(de)
options(scipen=5)
p3 <- ggplot(data=de, aes(x=CBS1,y=CBS2)) +  
      geom_point(shape = 21, colour = "black", fill="lightsteelblue3") +
      theme_bw() + theme(panel.grid.minor=element_blank(), panel.grid.major=element_blank()) + 
      scale_x_continuous(trans = 'log10') + scale_y_continuous(trans = 'log10')
      geom_text_repel(data=subset(de, de$ratio > 1.3 & de$CBS2 > 500), aes(label=id), size=3) +
      geom_point(data=subset(de, de$ratio > 1.3 & de$CBS2 > 500), col="red") +
      geom_text_repel(data=subset(de, de$ratio < 0.5 & de$CBS1 > 500), aes(label=id), size=3) +
      geom_point(data=subset(de, de$ratio < 0.5 & de$CBS1 > 500), col="red")

pdf("bulk.correlation.pdf", width=12, height=4, useDingbats=FALSE)
plot_grid(p1,p2,p3,ncol=3)
dev.off()

```

# Session Info

```{r sessinf}
sessionInfo()
```

