---
title: "SiT - Figures"
author: '[Kevin Lebrigand](mailto:lebrigand@ipmc.cnrs.fr)§, Joseph Bergenstråhle §, Kim Thrane §, Annelie Mollbrink, Pascal Barbry, Rainer Waldmann and Joakim Lundeberg'
date: 'Compiled: `r format(Sys.Date(), "%B %d, %Y")`'
abstract: 'The spatial landscape of gene expression isoforms in tissue sections.'
output:
  html_document:
    df_print: paged
    toc: yes
    toc_depth: '3'
  html_notebook:
    code_folding: none
    theme: journal
    toc: yes
    toc_depth: 3
    toc_float: yes
  pdf_document:
    toc: yes
    toc_depth: '3'
---

<style type="text/css">

body, td {
   font-size: 15px;
}
code.r{
  font-size: 15px;
}
pre {
  font-size: 15px
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  cache = FALSE,
  cache.lazy = FALSE,
  tidy = TRUE
)

```

```{r message=FALSE, warning=FALSE, echo=TRUE, eval=T, fig.height=4, fig.width=6}

setwd("thisdir")
source("00-common.import.R")

MOB <- readRDS("output/MOB.nanopore.deconv.rds")

```

# markers

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

pdf("figures/figures/Figure.main.2a1.pdf", width=8, height=6, useDingbats=FALSE)
SpatialDimPlot(MOB)
dev.off()

DefaultAssay(object = MOB) <- "Spatial"
#markers <- FindAllMarkers(object = MOB, only.pos = TRUE)
#top <- markers %>% group_by(cluster) %>% top_n(25, avg_log2FC)
#DoHeatmap(MOB, features=top$gene, size=3.5)
avg <- AverageExpression(MOB, return.seurat=TRUE, assays="Spatial")
#pheatmap(avg@assays$Spatial@scale.data[top$gene,], cluster_rows=FALSE, cluster_cols=FALSE, fontsize = 8, clustering_method = "complete")
genes <- c("Nrgn","Meis2","Mbp","Slc32a1","Bin1","Spp1","Ntng1","Shisa3","Slc17a7","Scn1b","Cnot3","Bc1","Slc1a2","Tatdn1","Lrrc17","Nrsn1","Th","Vsnl1","Shisa9","Eomes","Cldn5","Mgp","Apod","Apoe","Fabp7")

pdf("figures/Figure.main.2a2.pdf", width=8, height=8, useDingbats=FALSE)
pheatmap(avg@assays$Spatial@scale.data[genes,], cluster_rows=FALSE, cluster_cols=FALSE, fontsize = 8, clustering_method = "complete")
dev.off()

```

# Plp1

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

x <- unique(rownames(MOB@assays$ISO@data)[grep(paste("^Plp1..",sep=""), rownames(MOB@assays$ISO@data))])
DefaultAssay(MOB) <- "ISO"

pdf("figures/Figure.main.2c1.pdf", width=10, height=6, useDingbats=FALSE)
SpatialFeaturePlot(MOB, features = x, pt.size.factor=1.8, alpha = c(1, 1))
dev.off()


MOB[[x[1]]] <- MOB@assays$ISO@data[x[1],]
MOB[[x[2]]] <- MOB@assays$ISO@data[x[2],]

pdf("figures/Figure.main.2c2.pdf", width=6, height=6, useDingbats=FALSE)
SPOTlight::spatial_scatterpie(se_obj = MOB,
                              cell_types_all = x,
                              img_path = "data/MOB_tissue_lowres_image.png",
                              pie_scale = 0.5, scatterpie_alpha = 0.7)  +
                              scale_fill_manual(values = c("#cc3333","#ff9933"), breaks = x)
dev.off()

```

# Integration of external data (Tepe et al., Cell report, 2018) (GSE121891)

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

tepe <- readRDS("data/tepe.WT.rds")

# Plp1 expresser cell types
a <- aggregate(tepe@assays$RNA@counts["Plp1",], list(tepe@meta.data$ClusterName), sum)
a$nb <- as.data.frame(table(tepe@meta.data$ClusterName))$Freq
a$avg <- a$x/a$nb
plp1_cell_types <- as.character(a[a$avg>1,]$Group.1)

tepe[['ClusterNamePlp1']] <- tepe@meta.data$ClusterName
tepe <- SetIdent(tepe, value=tepe@meta.data$ClusterNamePlp1)
tepe[['ClusterNamePlp1']] <- tepe@active.ident
cells.use <- WhichCells(tepe, idents = plp1_cell_types, invert=TRUE)
tepe <- SetIdent(tepe, cells=cells.use, value="#NA")

mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(9)
mycolors[1] <- "#cfcfcf"
mycolors[2:9] <- colorRampPalette(brewer.pal(8, "Set1"))(8)

pdf("figures/Figure.main.2e.pdf", width=8, height=6, useDingbats=FALSE)
DimPlot(tepe, cols=mycolors, label=TRUE)
dev.off()

# ---------------------------
# Spotlight deconvolution
# --------------------------

#Seurat::Idents(object = tepe) <- tepe@meta.data$ClusterName
#cluster_markers_all <- Seurat::FindAllMarkers(object = tepe, 
#                                              assay = "SCT",
#                                              slot = "data",
#                                              verbose = TRUE, 
#                                              only.pos = TRUE)

#saveRDS(object = cluster_markers_all, file = here::here("markers_tepe.rds"))

#set.seed(123)
#spotlight_ls <- spotlight_deconvolution(
#  se_sc = tepe,
#  counts_spatial = MOB@assays$Spatial@counts,
#  clust_vr = "ClusterName", # Variable in sc_seu containing the cell-type annotation
#  cluster_markers = cluster_markers_all, # Dataframe with the marker genes
#  cl_n = 100, # number of cells per cell type to use
#  hvg = 3000, # Number of HVG to use
#  ntop = NULL, # How many of the marker genes to use (by default all)
#  transf = "uv", # Perform unit-variance scaling per cell and spot prior to factorzation and NLS
#  method = "nsNMF", # Factorization method
#  min_cont = 0 # Remove those cells contributing to a spot below a certain threshold 
#)

#saveRDS(object = spotlight_ls, file = here::here("spotlight_tepe.rds"))

#spotlight_ls <- readRDS("data/spotlight_tepe.rds")

#nmf_mod <- spotlight_ls[[1]]
#decon_mtrx <- spotlight_ls[[2]]

#h <- NMF::coef(nmf_mod[[1]])
#rownames(h) <- paste("Topic", 1:nrow(h), sep = "_")

#basis_spotlight <- data.frame(NMF::basis(nmf_mod[[1]]))

#colnames(basis_spotlight) <- unique(stringr::str_wrap(nmf_mod[[2]], width = 30))

#######################################################
# remove contributing cell types under 8% --> this is usefull for scatterpie spotlight plot not to have too many contributing cells
# we could keep all contributions for latter correlation computation, tested, improve a bit but message is the same so keep that standard filtering
# except for SiTx visualisation this gives much nicer representation
#######################################################
#decon_mtrx_sub <- decon_mtrx[, colnames(decon_mtrx) != "res_ss"]
#decon_mtrx_sub[decon_mtrx_sub < 0.08] <- 0
#decon_mtrx <- cbind(decon_mtrx_sub, "res_ss" = decon_mtrx[, "res_ss"])
#rownames(decon_mtrx) <- colnames(MOB)

#decon_df <- decon_mtrx %>%
#  data.frame() %>%
#  tibble::rownames_to_column("barcodes")

#MOB@meta.data <- MOB@meta.data %>%
#  tibble::rownames_to_column("barcodes") %>%
#  dplyr::left_join(decon_df, by = "barcodes") %>%
#  tibble::column_to_rownames("barcodes")


col=as.data.frame(ggplotColours(n = length(unique(tepe@meta.data$ClusterName))))
col$cell_type <- levels(tepe@meta.data$ClusterName)
colnames(col) <- c("color","celltype")
col

col <- col[col$celltype %in% plp1_cell_types,]

pdf("figures/Figure.main.2f1.pdf", width=8, height=6, useDingbats=FALSE)
SPOTlight::spatial_scatterpie(se_obj = MOB,
                              cell_types_all = plp1_cell_types,
                              img_path = "data/MOB_tissue_lowres_image.png",
                              pie_scale = 0.5, scatterpie_alpha = 2)  +
                              scale_fill_manual(values = col[, "color"], breaks = plp1_cell_types)
dev.off()

v <- numeric()
v2 <- numeric()
for(i in 1:length(plp1_cell_types)){
  df <- data.frame()
  x <- cor.test(MOB@meta.data[,plp1_cell_types[i]],MOB@assays$ISO@data["Plp1..ENSMUST00000033800.12",])$estimate
  x2 <- cor.test(MOB@meta.data[,plp1_cell_types[i]],MOB@assays$ISO@data["Plp1..ENSMUST00000113085.1",])$estimate
  v <- c(v, x)
  v2 <- c(v2, x2)
}
df <- as.data.frame(cbind(v,v2))
rownames(df) <- plp1_cell_types
colnames(df) <- c("Plp1-201","Plp1-202")

pheatmap(df, cluster_rows=TRUE, cluster_cols=FALSE, fontsize = 8, clustering_method = "complete", clustering_distance_rows = "manhattan")

df$id <- rownames(df)
df <- melt(df,id.vars = "id")

pdf("figures/Figure.main.2f2.pdf", width=16, height=10, useDingbats=FALSE)
ggplot(data=df, aes(x=id, y=value, fill=variable)) +
  geom_bar(stat="identity", position=position_dodge())+
  #geom_text(aes(label=variable), vjust=1.6, color="white",
  #          position = position_dodge(0.9), size=3.5)+
  scale_fill_manual(values=c("#ff0000", "#ff9015")) +
  theme_minimal()
dev.off()

```

# Session Info

```{r sessinf}
sessionInfo()
```

