---
title: "SiT - Figures"
author: '[Kevin Lebrigand](mailto:lebrigand@ipmc.cnrs.fr)§, Joseph Bergenstråhle §, Kim Thrane §, Annelie Mollbrink, Pascal Barbry, Rainer Waldmann and Joakim Lundeberg'
date: 'Compiled: `r format(Sys.Date(), "%B %d, %Y")`'
abstract: 'The spatial landscape of gene expression isoforms in tissue sections.'
output:
  html_document:
    df_print: paged
    toc: yes
    toc_depth: '3'
  html_notebook:
    code_folding: none
    theme: journal
    toc: yes
    toc_depth: 3
    toc_float: yes
  pdf_document:
    toc: yes
    toc_depth: '3'
---

<style type="text/css">

body, td {
   font-size: 15px;
}
code.r{
  font-size: 15px;
}
pre {
  font-size: 15px
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  cache = FALSE,
  cache.lazy = FALSE,
  tidy = TRUE
)

```

```{r message=FALSE, warning=FALSE, echo=TRUE, eval=T, fig.height=4, fig.width=6}

setwd("thisdir")
source("00-common.import.R")

CBS2 <- readRDS("output/CBS2.nanopore.deconv.rds")
CBS1 <- readRDS("output/CBS1.nanopore.deconv.rds")
DefaultAssay(CBS1) <- "ISO"
DefaultAssay(CBS2) <- "ISO"

clusters <- c("Midbrain","Hippocampus area","Isocortex-1","Isocortex-2","Olfactory area","Fiber tracts","Retrosplenial area","CA1/CA2","Thalamus","Hypothalamus","CA3","DG")

```

# Caly

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

x <- unique(rownames(CBS2@assays$ISO@data)[grep(paste("^Caly..",sep=""), rownames(CBS2@assays$ISO@data))])
x <- c("Caly..ENSMUST00000211044.1","Caly..ENSMUST00000211283.1")
p1 <- SpatialFeaturePlot(CBS1, features = x, pt.size.factor=1.8, alpha = c(1, 1))
p2 <- SpatialFeaturePlot(CBS2, features = x, pt.size.factor=1.8, alpha = c(1, 1))

a.2 <- aggregate(CBS2@assays$ISO@counts["Caly..ENSMUST00000211044.1",], list(CBS2@active.ident), sum)
b.2 <- aggregate(CBS2@assays$ISO@counts["Caly..ENSMUST00000211283.1",], list(CBS2@active.ident), sum)
a.1 <- aggregate(CBS1@assays$ISO@counts["Caly..ENSMUST00000211044.1",], list(CBS1@active.ident), sum)
b.1 <- aggregate(CBS1@assays$ISO@counts["Caly..ENSMUST00000211283.1",], list(CBS1@active.ident), sum)
a.1 <- a.1[-c(13),]
b.1 <- b.1[-c(13),]

caly <- cbind(a.1,b.1$x,a.2$x,b.2$x)
colnames(caly) <- c("region","CBS1.203","CBS1.204","CBS2.203","CBS2.204")
caly$CBS1.ratio.203 <- (100*caly$CBS1.203)/(caly$CBS1.203+caly$CBS1.204)
caly$CBS2.ratio.203 <- (100*caly$CBS2.203)/(caly$CBS2.203+caly$CBS2.204)
caly$CBS1.ratio.204 <- (100*caly$CBS1.204)/(caly$CBS1.203+caly$CBS1.204)
caly$CBS2.ratio.204 <- (100*caly$CBS2.204)/(caly$CBS2.203+caly$CBS2.204)
caly <- caly[,-c(2:5)]
df <- melt(caly, id.vars="region")

pdf("figures/Figure.sup.11.caly.pdf", width=10, height=6, useDingbats=FALSE)
plot_grid(p1,p2)
ggplot(df, aes(x=region, y=value, fill=variable)) +
    geom_bar(stat='identity', position='dodge') +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
    scale_fill_manual(values=c("#ff0000","#ff0000", "#ff9015","#ff9015")) 
dev.off()  

```

# Cnih2

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

x <- unique(rownames(CBS2@assays$ISO@data)[grep(paste("^Cnih2..",sep=""), rownames(CBS2@assays$ISO@data))])
p1 <- SpatialFeaturePlot(CBS1, features = x, pt.size.factor=1.8, alpha = c(1, 1))
p2 <- SpatialFeaturePlot(CBS2, features = x, pt.size.factor=1.8, alpha = c(1, 1))

a.2 <- aggregate(CBS2@assays$ISO@counts["Cnih2..ENSMUST00000025805.7",], list(CBS2@active.ident), sum)
b.2 <- aggregate(CBS2@assays$ISO@counts["Cnih2..ENSMUST00000235638.1",], list(CBS2@active.ident), sum)
a.1 <- aggregate(CBS1@assays$ISO@counts["Cnih2..ENSMUST00000025805.7",], list(CBS1@active.ident), sum)
b.1 <- aggregate(CBS1@assays$ISO@counts["Cnih2..ENSMUST00000235638.1",], list(CBS1@active.ident), sum)
a.1 <- a.1[-c(13),]
b.1 <- b.1[-c(13),]

cnih2 <- cbind(a.1,b.1$x,a.2$x,b.2$x)
colnames(cnih2) <- c("region","CBS1.201","CBS1.202","CBS2.201","CBS2.202")
cnih2$CBS1.ratio.201 <- (100*cnih2$CBS1.201)/(cnih2$CBS1.201+cnih2$CBS1.202)
cnih2$CBS2.ratio.201 <- (100*cnih2$CBS2.201)/(cnih2$CBS2.201+cnih2$CBS2.202)
cnih2$CBS1.ratio.202 <- (100*cnih2$CBS1.202)/(cnih2$CBS1.201+cnih2$CBS1.202)
cnih2$CBS2.ratio.202 <- (100*cnih2$CBS2.202)/(cnih2$CBS2.201+cnih2$CBS2.202)
cnih2 <- cnih2[,-c(2:5)]
df <- melt(cnih2, id.vars="region")

pdf("figures/Figure.sup.11.cnih2.pdf", width=10, height=6, useDingbats=FALSE)
plot_grid(p1,p2)
ggplot(df, aes(x=region, y=value, fill=variable)) +
    geom_bar(stat='identity', position='dodge') +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
    scale_fill_manual(values=c("#ff0000","#ff0000", "#ff9015","#ff9015")) 
dev.off()  

```

# Dtnbp1

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

x <- unique(rownames(CBS2@assays$ISO@data)[grep(paste("^Dtnbp1..",sep=""), rownames(CBS2@assays$ISO@data))])
x <- c("Dtnbp1..ENSMUST00000222321.1","Dtnbp1..ENSMUST00000072329.14")
p1 <- SpatialFeaturePlot(CBS1, features = x, pt.size.factor=1.8, alpha = c(1, 1))
p2 <- SpatialFeaturePlot(CBS2, features = x, pt.size.factor=1.8, alpha = c(1, 1))

a.2 <- aggregate(CBS2@assays$ISO@counts["Dtnbp1..ENSMUST00000222321.1",], list(CBS2@active.ident), sum)
b.2 <- aggregate(CBS2@assays$ISO@counts["Dtnbp1..ENSMUST00000072329.14",], list(CBS2@active.ident), sum)
a.1 <- aggregate(CBS1@assays$ISO@counts["Dtnbp1..ENSMUST00000222321.1",], list(CBS1@active.ident), sum)
b.1 <- aggregate(CBS1@assays$ISO@counts["Dtnbp1..ENSMUST00000072329.14",], list(CBS1@active.ident), sum)
a.1 <- a.1[-c(13),]
b.1 <- b.1[-c(13),]

dtnbp1 <- cbind(a.1,b.1$x,a.2$x,b.2$x)
colnames(dtnbp1) <- c("region","CBS1.206","CBS1.201","CBS2.206","CBS2.201")
dtnbp1$CBS1.ratio.201 <- (100*dtnbp1$CBS1.201)/(dtnbp1$CBS1.201+dtnbp1$CBS1.206)
dtnbp1$CBS2.ratio.201 <- (100*dtnbp1$CBS2.201)/(dtnbp1$CBS2.201+dtnbp1$CBS2.206)
dtnbp1$CBS1.ratio.206 <- (100*dtnbp1$CBS1.206)/(dtnbp1$CBS1.201+dtnbp1$CBS1.206)
dtnbp1$CBS2.ratio.206 <- (100*dtnbp1$CBS2.206)/(dtnbp1$CBS2.201+dtnbp1$CBS2.206)
dtnbp1 <- dtnbp1[,-c(2:5)]
df <- melt(dtnbp1, id.vars="region")

pdf("figures/Figure.sup.11.dtnbp1.pdf", width=10, height=6, useDingbats=FALSE)
plot_grid(p1,p2)
ggplot(df, aes(x=region, y=value, fill=variable)) +
    geom_bar(stat='identity', position='dodge') +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
    scale_fill_manual(values=c("#ff0000","#ff0000", "#ff9015","#ff9015")) 
dev.off()  

```

# Aldoa

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

x <- unique(rownames(CBS2@assays$ISO@data)[grep(paste("^Aldoa..",sep=""), rownames(CBS2@assays$ISO@data))])
x <- c("Aldoa..ENSMUST00000106348.7","Aldoa..ENSMUST00000032934.11")
p1 <- SpatialFeaturePlot(CBS1, features = x, pt.size.factor=1.8, alpha = c(1, 1))
p2 <- SpatialFeaturePlot(CBS2, features = x, pt.size.factor=1.8, alpha = c(1, 1))

a.2 <- aggregate(CBS2@assays$ISO@counts["Aldoa..ENSMUST00000032934.11",], list(CBS2@active.ident), sum)
b.2 <- aggregate(CBS2@assays$ISO@counts["Aldoa..ENSMUST00000106348.7",], list(CBS2@active.ident), sum)
a.1 <- aggregate(CBS1@assays$ISO@counts["Aldoa..ENSMUST00000032934.11",], list(CBS1@active.ident), sum)
b.1 <- aggregate(CBS1@assays$ISO@counts["Aldoa..ENSMUST00000106348.7",], list(CBS1@active.ident), sum)
a.1 <- a.1[-c(13),]
b.1 <- b.1[-c(13),]

aldoa <- cbind(a.1,b.1$x,a.2$x,b.2$x)
colnames(aldoa) <- c("region","CBS1.201","CBS1.203","CBS2.201","CBS2.203")
aldoa$CBS1.ratio.201 <- (100*aldoa$CBS1.201)/(aldoa$CBS1.201+aldoa$CBS1.203)
aldoa$CBS2.ratio.201 <- (100*aldoa$CBS2.201)/(aldoa$CBS2.201+aldoa$CBS2.203)
aldoa$CBS1.ratio.203 <- (100*aldoa$CBS1.203)/(aldoa$CBS1.201+aldoa$CBS1.203)
aldoa$CBS2.ratio.203 <- (100*aldoa$CBS2.203)/(aldoa$CBS2.201+aldoa$CBS2.203)
aldoa <- aldoa[,-c(2:5)]
df <- melt(aldoa, id.vars="region")

pdf("figures/Figure.sup.11.aldoa.pdf", width=10, height=6, useDingbats=FALSE)
plot_grid(p1,p2)
ggplot(df, aes(x=region, y=value, fill=variable)) +
    geom_bar(stat='identity', position='dodge') +
    theme_minimal() +
    theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
    scale_fill_manual(values=c("#ff0000","#ff0000", "#ff9015","#ff9015")) 
dev.off()  


```

# Session Info

```{r sessinf}
sessionInfo()
```

