---
title: "SiT - Figures"
author: '[Kevin Lebrigand](mailto:lebrigand@ipmc.cnrs.fr)§, Joseph Bergenstråhle §, Kim Thrane §, Annelie Mollbrink, Pascal Barbry, Rainer Waldmann and Joakim Lundeberg'
date: 'Compiled: `r format(Sys.Date(), "%B %d, %Y")`'
abstract: 'The spatial landscape of gene expression isoforms in tissue sections.'
output:
  html_document:
    df_print: paged
    toc: yes
    toc_depth: '3'
  html_notebook:
    code_folding: none
    theme: journal
    toc: yes
    toc_depth: 3
    toc_float: yes
  pdf_document:
    toc: yes
    toc_depth: '3'
---

<style type="text/css">

body, td {
   font-size: 15px;
}
code.r{
  font-size: 15px;
}
pre {
  font-size: 15px
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  cache = FALSE,
  cache.lazy = FALSE,
  tidy = TRUE
)

```

```{r message=FALSE, warning=FALSE, echo=TRUE, eval=T, fig.height=4, fig.width=6}

setwd("/data/10x_data/10x_visium/000-github/")
source("00.common.import.R")

MOB <- readRDS("./output/MOB.nanopore.rds")

```

# Integration of external data (Tepe et al., Cell report, 2018) (GSE121891)

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

#raw_counts<-read.table(file=paste0("/data/10x_data/10x_visium/final/ressources/","GSE121891_OB_6_runs_processed_seurat.dge.csv"),sep=",")
#raw_counts[1:5,1:5]
#metadata<-read.table(file=paste0("/data/10x_data/10x_visium/final/ressources/","GSE121891_OB_metaData_seurat.csv"),sep=",", header=TRUE)
#rownames(metadata) <- metadata$X
#metadata <- metadata[,-c(1)]
#metadata[1:5,1:7]
#gse <- CreateSeuratObject(raw_counts, project = "GSE121891", assay = "RNA", meta.data = metadata)
#gse <- SCTransform(gse, ncells = 3000, verbose = FALSE) %>% RunPCA(verbose = FALSE) %>% RunUMAP(dims = 1:30)
#saveRDS(gse, "./output/tepe.rds")

tepe <- readRDS("../output/tepe.rds")
tepe <- subset(tepe, subset = orig.ident == "WT1" | orig.ident == "WT2")

p1 <- DimPlot(tepe, group.by = "ClusterName", label=TRUE)
p1
#DefaultAssay(MOB) <- "SCT"
#anchors <- FindTransferAnchors(reference = tepe, query = MOB, normalization.method = "SCT")
#predictions.assay <- TransferData(anchorset = anchors, refdata = tepe$ClusterName, prediction.assay = TRUE, weight.reduction = MOB[["pca"]], dims = 1:30)
#MOB[["predictions"]] <- predictions.assay

#DefaultAssay(MOB) <- "predictions"
#DotPlot(MOB, features = sort(rownames(MOB@assays$predictions@data))) + RotatedAxis()

DefaultAssay(tepe) <- "RNA"
VlnPlot(tepe,feature="Plp1", group.by="ClusterName")
p2 <- DotPlot(tepe,feature="Plp1", group.by="ClusterName")

#DefaultAssay(tepe) <- "RNA"
#SpatialFeaturePlot(MOB, features = c("OPC","OEC1","OEC2","OEC3","OEC4","OEC5","N2","MyOligo"), pt.size.factor = 2.5, ncol = 3, alpha = c(0.1, 1))

# ---------------------------
# give a try to spotlight
# --------------------------

#Seurat::Idents(object = tepe) <- tepe@meta.data$ClusterName
#cluster_markers_all <- Seurat::FindAllMarkers(object = tepe, 
#                                              assay = "SCT",
#                                              slot = "data",
#                                              verbose = TRUE, 
#                                              only.pos = TRUE)

#saveRDS(object = cluster_markers_all, file = here::here("markers_tepe.rds"))

#set.seed(123)
#spotlight_ls <- spotlight_deconvolution(
#  se_sc = tepe,
#  counts_spatial = MOB@assays$Spatial@counts,
#  clust_vr = "ClusterName", # Variable in sc_seu containing the cell-type annotation
#  cluster_markers = cluster_markers_all, # Dataframe with the marker genes
#  cl_n = 100, # number of cells per cell type to use
#  hvg = 3000, # Number of HVG to use
#  ntop = NULL, # How many of the marker genes to use (by default all)
#  transf = "uv", # Perform unit-variance scaling per cell and spot prior to factorzation and NLS
#  method = "nsNMF", # Factorization method
#  min_cont = 0 # Remove those cells contributing to a spot below a certain threshold 
#)

#saveRDS(object = spotlight_ls, file = here::here("spotlight_tepe.rds"))
spotlight_ls <- readRDS("spotlight_tepe.rds")

nmf_mod <- spotlight_ls[[1]]
decon_mtrx <- spotlight_ls[[2]]

h <- NMF::coef(nmf_mod[[1]])
rownames(h) <- paste("Topic", 1:nrow(h), sep = "_")
topic_profile_plts <- SPOTlight::dot_plot_profiles_fun(
  h = h,
  train_cell_clust = nmf_mod[[2]])

topic_profile_plts[[2]] + ggplot2::theme(
  axis.text.x = ggplot2::element_text(angle = 90), 
  axis.text = ggplot2::element_text(size = 12))

topic_profile_plts[[1]] + theme(axis.text.x = element_text(angle = 90), 
                                axis.text = element_text(size = 12))

basis_spotlight <- data.frame(NMF::basis(nmf_mod[[1]]))

colnames(basis_spotlight) <- unique(stringr::str_wrap(nmf_mod[[2]], width = 30))

basis_spotlight %>%
  dplyr::arrange(desc(MyOligo)) %>%
  round(., 5) %>% 
  DT::datatable(., filter = "top")


decon_mtrx_sub <- decon_mtrx[, colnames(decon_mtrx) != "res_ss"]
decon_mtrx_sub[decon_mtrx_sub < 0.08] <- 0
decon_mtrx <- cbind(decon_mtrx_sub, "res_ss" = decon_mtrx[, "res_ss"])
rownames(decon_mtrx) <- colnames(MOB)

decon_df <- decon_mtrx %>%
  data.frame() %>%
  tibble::rownames_to_column("barcodes")

MOB@meta.data <- MOB@meta.data %>%
  tibble::rownames_to_column("barcodes") %>%
  dplyr::left_join(decon_df, by = "barcodes") %>%
  tibble::column_to_rownames("barcodes")

Seurat::SpatialFeaturePlot(
  object = MOB,
  features = c("OPC","OEC1","OEC2","OEC3","OEC4","OEC5","MyOligo", "N2"),
  alpha = c(0.1, 1),
  pt.size.factor = 2.5)

cell_types_all <- colnames(decon_mtrx)[which(colnames(decon_mtrx) != "res_ss")]

pdf("spotlight.pdf", width=12, height=12, useDingbats=FALSE)
SPOTlight::spatial_scatterpie(se_obj = MOB,
                              cell_types_all = cell_types_all,
                              img_path = "/data/10x_data/10x_visium/illumina/MOB/spatial/tissue_lowres_image.png",
                              pie_scale = 0.6)
dev.off()

col=as.data.frame(ggplotColours(n = length(unique(tepe@meta.data$ClusterName))))
col$cell_type <- levels(tepe@meta.data$ClusterName)
colnames(col) <- c("color","celltype")
col

cell_types <- c("MyOligo","N2","OEC1","OEC2","OEC3","OEC4","OEC5","OPC")
col <- col[col$celltype %in% cell_types,]

p3 <- SPOTlight::spatial_scatterpie(se_obj = MOB,
                              cell_types_all = cell_types,
                              img_path = "/data/10x_data/10x_visium/illumina/MOB/spatial/tissue_lowres_image.png",
                              pie_scale = 0.5, scatterpie_alpha = 2)  +
                              scale_fill_manual(values = col[, "color"], breaks = cell_types)


pdf("spotlight_MOB.pdf", width=24, height=10, useDingbats=FALSE)
plot_grid(p1,p2,p3,ncol=3)
dev.off()

```

# Session Info

```{r sessinf}
sessionInfo()
```

