---
title: "SiT - Short read / Long read classification"
author: '[Kevin Lebrigand](mailto:lebrigand@ipmc.cnrs.fr)§, Joseph Bergenstråhle §, Kim Thrane §, Annelie Mollbrink, Pascal Barbry, Rainer Waldmann and Joakim Lundeberg'
date: 'Compiled: `r format(Sys.Date(), "%B %d, %Y")`'
abstract: 'The spatial landscape of gene expression isoforms in tissue sections.'
output:
  html_document:
    df_print: paged
    toc: yes
    toc_depth: '3'
  html_notebook:
    code_folding: none
    theme: journal
    toc: yes
    toc_depth: 3
    toc_float: yes
  pdf_document:
    toc: yes
    toc_depth: '3'
---

<style type="text/css">

body, td {
   font-size: 15px;
}
code.r{
  font-size: 15px;
}
pre {
  font-size: 15px
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  cache = FALSE,
  cache.lazy = FALSE,
  tidy = TRUE
)
```

# Loading Spatial data

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=4, fig.width=6}

setwd("/data/10x_data/10x_visium/")
source("00.import.R")

CBS2 <- Load10X_Spatial(data.dir = "/data/10x_data/000-notebooks/visium/142079-spatial-hippo")

```

# Quality control, no filtration

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=16}

mito.genes <- grep(pattern = "^mt-", x = rownames(CBS2@assays$Spatial), value = TRUE)
dropouts <- Matrix::colSums(CBS2@assays$Spatial@data == 0)/nrow(CBS2@assays$Spatial)
ribo.genes <- grep(pattern = "^Rp[sl]", x = rownames(CBS2@assays$Spatial), value = TRUE)
percent.mito <- Matrix::colSums(CBS2@assays$Spatial[mito.genes, ])/Matrix::colSums(CBS2@assays$Spatial)
percent.ribo <- Matrix::colSums(CBS2@assays$Spatial[ribo.genes, ])/Matrix::colSums(CBS2@assays$Spatial)
CBS2[['percent.mito']] <- percent.mito
CBS2[['percent.ribo']] <- percent.ribo
CBS2[['dropouts']] <- dropouts

VlnPlot(CBS2, features = c("nFeature_Spatial", "nCount_Spatial","dropouts","percent.ribo","percent.mito"), ncol=5, cols = "lightsteelblue3")

```

# Add Nanopore data: ISO, ISOG assays

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=6, fig.width=8}

all = read.delim("./nanopore/CBS2_genematrix.txt", stringsAsFactors = F)
data = all[,2:ncol(all)]
colnames(data) <- paste(colnames(data),"-1", sep="")
rownames(data) <- all$geneId
vect <- colnames(CBS2@assays$Spatial)
data <- data[vect]
CBS2[["ISOG"]] <- CreateAssayObject(counts = data)

all = read.delim("./nanopore/CBS2_isomatrix.txt", stringsAsFactors = F)
data = all[,4:ncol(all)]
dim(data)
colnames(data) <- paste(colnames(data),"-1", sep="")
rownames(data) = paste(all$geneId, all$transcriptId, sep="..")
vect <- colnames(CBS2@assays$Spatial)
idx <- grep("undef", rownames(data), invert = TRUE)
data <- data[idx,vect]
dim(data)
CBS2[["ISO"]] <- CreateAssayObject(counts = data)

```

# Spatial analysis

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=4, fig.width=6}

sp <- SCTransform(CBS2, assay = "Spatial", verbose = FALSE)
sp <- RunPCA(sp, assay = "SCT", verbose = FALSE)
sp <- RunUMAP(sp, reduction = "pca", dims = 1:30)
sp <- FindNeighbors(sp, reduction = "pca", dims = 1:30)
sp <- FindClusters(sp, verbose = FALSE, resolution = 0.4)

p1 <- SpatialDimPlot(sp, label = TRUE, label.size = 3, pt.size.factor = 1.6) + ggtitle("spatial")
p1

```

# ISOG analysis

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=4, fig.width=6}

isog <- SCTransform(CBS2, assay = "ISOG", verbose = FALSE)
isog <- RunPCA(isog, assay = "SCT", verbose = FALSE)
isog <- RunUMAP(isog, reduction = "pca", dims = 1:30)
isog <- FindNeighbors(isog, reduction = "pca", dims = 1:30)
isog <- FindClusters(isog, verbose = FALSE, resolution = 0.4)

p2 <- SpatialDimPlot(isog, label = TRUE, label.size = 3, pt.size.factor = 1.6) + ggtitle("ISOG")
p2

```

# ISO analysis

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=4, fig.width=6}

iso <- SCTransform(CBS2, assay = "ISO", verbose = FALSE)
iso <- RunPCA(iso, assay = "SCT", verbose = FALSE)
iso <- RunUMAP(iso, reduction = "pca", dims = 1:30)
iso <- FindNeighbors(iso, reduction = "pca", dims = 1:30)
iso <- FindClusters(iso, verbose = FALSE, resolution = 0.4)

p3 <- SpatialDimPlot(iso, label = TRUE, label.size = 3, pt.size.factor = 1.6) + ggtitle("ISO")
p3

```

# Comparison

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=4, fig.width=12}

plot_grid(p1,p2,p3, ncol=3)

tiff("figures/classification.tif", units="in", width=14, height=6, res=300)
plot_grid(p1,p2,p3, ncol=3)
dev.off()

```

# Session Info

```{r sessinf}
sessionInfo()
```

