---
title: "SiT - Isoforms focus"
author: '[Kevin Lebrigand](mailto:lebrigand@ipmc.cnrs.fr)§, Joseph Bergenstråhle §, Kim Thrane §, Annelie Mollbrink, Pascal Barbry, Rainer Waldmann and Joakim Lundeberg'
date: 'Compiled: `r format(Sys.Date(), "%B %d, %Y")`'
abstract: 'The spatial landscape of gene expression isoforms in tissue sections.'
output:
  html_document:
    df_print: paged
    toc: yes
    toc_depth: '3'
  html_notebook:
    code_folding: none
    theme: journal
    toc: yes
    toc_depth: 3
    toc_float: yes
  pdf_document:
    toc: yes
    toc_depth: '3'
---

<style type="text/css">

body, td {
   font-size: 15px;
}
code.r{
  font-size: 15px;
}
pre {
  font-size: 15px
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  cache = FALSE,
  cache.lazy = FALSE,
  tidy = TRUE
)

```

# Loading CBS1 and CBS2

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=4, fig.width=6}

setwd("/data/10x_data/10x_visium/")
source("00.import.R")

CBS1 <- readRDS("./output/CBS1.nanopore.rds")
CBS2 <- readRDS("./output/CBS2.nanopore.rds")
MOB <- readRDS("./output/MOB.nanopore.rds")

avg.iso.CBS1 <- AverageExpression(object = CBS1, return.seurat = TRUE, assay="ISO")
avg.iso.CBS1 <- scale_my_data(avg.iso.CBS1, assay="ISO")
avg.iso.CBS2 <- AverageExpression(object = CBS2, return.seurat = TRUE, assay="ISO")
avg.iso.CBS2 <- scale_my_data(avg.iso.CBS2, assay="ISO")

```

## Isoform switch in CBS2

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=4, fig.width=8 }

DefaultAssay(object = CBS2) <- "ISO"
general <- c('Aldoa','Arl2bp','Caly','Capzb','Cdc42','Clta','Cltb','Cnih2','Dbndd2','Dtnbp1','Faim','Ftl1','Gnas','Kctd13','Kctd17','Ly6h','Mrpl48','Myl6','Nbdy','Nkain4','Olfm2','Rps6','Rtn4','Sept5','Sft2d1','Snap25','Vti1b')

lst <- c()
for (i in (1:length(general))){
  x <- rownames(CBS2@assays$ISO@counts)[grep(paste("^",general[i],"\\.\\.",sep=""), rownames(CBS2@assays$ISO@counts))]
  lst <- c(lst,x)
}

iso.switching = read.delim("./output/CBS2.isoswitch.top.csv", sep=",", stringsAsFactors = F)
iso.switching$key <- paste(iso.switching$geneId,iso.switching$transcriptId, sep="..")
lst <- lst[lst %in% iso.switching$key == TRUE]
DotPlot(object = CBS2, features = lst, scale.max = 100) + theme(axis.text.x = element_text(angle = 90, hjust = 1)) # RotatedAxis

DefaultAssay(object = CBS2) <- "Spatial"
DotPlot(object = CBS2, features = general, scale.max = 100, cols = c("lightgrey", "red")) + theme(axis.text.x = element_text(angle = 90, hjust = 1))

pdf("figures/isoswitch.pdf", width=36, height=8, useDingbats=FALSE)
DefaultAssay(object = CBS2) <- "ISO"
DotPlot(object = CBS2, features = lst, scale.max = 100) + theme(axis.text.x = element_text(angle = 45, hjust = 1))
DefaultAssay(object = CBS2) <- "Spatial"
print(DotPlot(object = CBS2, features = general, scale.max = 100, cols = c("lightgrey", "red")) + theme(axis.text.x = element_text(angle = 45, hjust = 1))) # 
dev.off()

```

## Snap25

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=4, fig.width=8 }

DefaultAssay(object = CBS1) <- "Spatial"
DefaultAssay(object = CBS2) <- "SCT"
SpatialFeaturePlot(CBS2, features = "Snap25", alpha = c(0.1, 1), pt.size.factor =2)

x <- unique(rownames(CBS1@assays$ISO@scale.data)[grep(paste("^Snap25\\.\\.",sep=""), rownames(CBS1@assays$ISO@scale.data))])
x <- c("Snap25..ENSMUST00000110098.3","Snap25..ENSMUST00000028727.10")
SpatialFeaturePlot(CBS1, features = x, alpha = c(0.1, 1), pt.size.factor =2)
SpatialFeaturePlot(CBS2, features = x, alpha = c(0.1, 1), pt.size.factor =2)

DotPlot(CBS1, features = x) + RotatedAxis() + theme(text = element_text(size = 8), axis.text=element_text(size = 8))
DotPlot(CBS2, features = x) + RotatedAxis() + theme(text = element_text(size = 8), axis.text=element_text(size = 8))
#avg.junc.scale <- AverageExpression(object = CBS1, return.seurat = TRUE, assay="JUNC")
#avg.junc.scale <- scale_my_data(avg.junc.scale, assay="JUNC")

pheatmap(avg.iso.CBS1@assays$ISO@scale.data[sort(x, decreasing=FALSE),], cluster_rows=FALSE, cluster_cols=FALSE, fontsize = 8, clustering_method = "complete", clustering_distance_rows = "manhattan")
pheatmap(avg.iso.CBS2@assays$ISO@scale.data[sort(x, decreasing=FALSE),], cluster_rows=FALSE, cluster_cols=FALSE, fontsize = 8, clustering_method = "complete", clustering_distance_rows = "manhattan")

```

## Gnas

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=4, fig.width=8 }

x <- unique(rownames(CBS1@assays$ISO@scale.data)[grep(paste("^Gnas\\.\\.",sep=""), rownames(CBS1@assays$ISO@scale.data))])
x <- c("Gnas..ENSMUST00000109087.7","Gnas..ENSMUST00000156623.7")
SpatialFeaturePlot(CBS1, features = x, alpha = c(0.1, 1), pt.size.factor =2)
SpatialFeaturePlot(CBS2, features = x, alpha = c(0.1, 1), pt.size.factor =2)

DotPlot(CBS1, features = x) + RotatedAxis() + theme(text = element_text(size = 8), axis.text=element_text(size = 8))
DotPlot(CBS2, features = x) + RotatedAxis() + theme(text = element_text(size = 8), axis.text=element_text(size = 8))
#avg.junc.scale <- AverageExpression(object = CBS1, return.seurat = TRUE, assay="JUNC")
#avg.junc.scale <- scale_my_data(avg.junc.scale, assay="JUNC")

pheatmap(avg.iso.CBS1@assays$ISO@scale.data[sort(x, decreasing=FALSE),], cluster_rows=FALSE, cluster_cols=FALSE, fontsize = 8, clustering_method = "complete", clustering_distance_rows = "manhattan")
pheatmap(avg.iso.CBS2@assays$ISO@scale.data[sort(x, decreasing=FALSE),], cluster_rows=FALSE, cluster_cols=FALSE, fontsize = 8, clustering_method = "complete", clustering_distance_rows = "manhattan")

CBS1@meta.data$gnas_ratio <- abs(CBS1@assays$ISO@scale.data["Gnas..ENSMUST00000109087.7",]/CBS1@assays$ISO@scale.data["Gnas..ENSMUST00000156623.7",])
p1 <- VlnPlot(CBS1,x)
CBS2@meta.data$gnas_ratio <- abs(CBS2@assays$ISO@scale.data["Gnas..ENSMUST00000109087.7",]/CBS2@assays$ISO@scale.data["Gnas..ENSMUST00000156623.7",])
p2 <- VlnPlot(CBS2,x)
plot_grid(p1,p2)

```

## Bin1

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=4, fig.width=8 }

x <- unique(rownames(CBS1@assays$ISO@scale.data)[grep(paste("^Bin1\\.\\.",sep=""), rownames(CBS1@assays$ISO@scale.data))])
x <- c("Bin1..ENSMUST00000234496.1","Bin1..ENSMUST00000025239.8")
SpatialFeaturePlot(CBS1, features = x, alpha = c(0.1, 1), pt.size.factor =2)
SpatialFeaturePlot(CBS2, features = x, alpha = c(0.1, 1), pt.size.factor =2)

DotPlot(CBS1, features = x) + RotatedAxis() + theme(text = element_text(size = 8), axis.text=element_text(size = 8))
DotPlot(CBS2, features = x) + RotatedAxis() + theme(text = element_text(size = 8), axis.text=element_text(size = 8))
#avg.junc.scale <- AverageExpression(object = CBS1, return.seurat = TRUE, assay="JUNC")
#avg.junc.scale <- scale_my_data(avg.junc.scale, assay="JUNC")

pheatmap(avg.iso.CBS1@assays$ISO@scale.data[sort(x, decreasing=FALSE),], cluster_rows=FALSE, cluster_cols=FALSE, fontsize = 8, clustering_method = "complete", clustering_distance_rows = "manhattan")
pheatmap(avg.iso.CBS2@assays$ISO@scale.data[sort(x, decreasing=FALSE),], cluster_rows=FALSE, cluster_cols=FALSE, fontsize = 8, clustering_method = "complete", clustering_distance_rows = "manhattan")

CBS1@meta.data$bin1_ratio <- abs(CBS1@assays$ISO@scale.data["Bin1..ENSMUST00000234496.1",]/CBS1@assays$ISO@scale.data["Bin1..ENSMUST00000025239.8",])
p1 <- VlnPlot(CBS1,"bin1_ratio")
CBS2@meta.data$bin1_ratio <- abs(CBS2@assays$ISO@scale.data["Bin1..ENSMUST00000234496.1",]/CBS2@assays$ISO@scale.data["Bin1..ENSMUST00000025239.8",])
p2 <- VlnPlot(CBS2,"bin1_ratio")
plot_grid(p1,p2)

```

## Plp1

```{r message=FALSE, warning=FALSE, echo=FALSE, eval=T, fig.height=4, fig.width=8 }

DefaultAssay(MOB) <- "Spatial"
x <- unique(rownames(MOB@assays$ISO@scale.data)[grep(paste("^Plp1\\.\\.",sep=""), rownames(MOB@assays$ISO@scale.data))])
SpatialFeaturePlot(MOB, features = x, alpha = c(0.1, 1), pt.size.factor =2)

DefaultAssay(MOB) <- "SCT"
SpatialFeaturePlot(MOB, features = "Plp1", alpha = c(0.1, 1), pt.size.factor =2)

DefaultAssay(MOB) <- "SCT"
SpatialFeaturePlot(MOB, features = "Plp1", alpha = c(0.1, 1), pt.size.factor =2)

DotPlot(MOB, features = x) + RotatedAxis() + theme(text = element_text(size = 8), axis.text=element_text(size = 8))
#avg.junc.scale <- AverageExpression(object = CBS1, return.seurat = TRUE, assay="JUNC")
#avg.junc.scale <- scale_my_data(avg.junc.scale, assay="JUNC")

pheatmap(avg.iso.CBS1@assays$ISO@scale.data[sort(x, decreasing=FALSE),], cluster_rows=FALSE, cluster_cols=FALSE, fontsize = 8, clustering_method = "complete", clustering_distance_rows = "manhattan")
pheatmap(avg.iso.CBS2@assays$ISO@scale.data[sort(x, decreasing=FALSE),], cluster_rows=FALSE, cluster_cols=FALSE, fontsize = 8, clustering_method = "complete", clustering_distance_rows = "manhattan")

MOB@meta.data$plp1_ratio <- MOB@assays$ISO@scale.data["Plp1..ENSMUST00000033800.12",]/MOB@assays$ISO@scale.data["Plp1..ENSMUST00000113085.1",]
VlnPlot(MOB,"plp1_ratio")

```

# Session Info

```{r sessinf}
sessionInfo()
```

